%!TEX root = main.tex

\section*{Introduction}

Genetic markers have long been used in the 
management of Pacific salmon, and salmon conservation and management has been
a driving force behind a number of advances in molecular ecology,
from data generation techniques \citep{clemento2011discovery,campbell2015genotyping,mckinney2017managing}
to statistical methodology
\citep{smouse1990genetic,anderson2002model,pella2006gibbs}.
Two broadly applicable innovations that have been
actively fostered by the Pacific salmon research community are genetic
stock identification
(GSI:~\citealt{milner1982genetic,beacham2004dna,seeb2007development})
and parentage based tagging
(PBT:~\citealt{anderson2006power, garza2007large, abadia2013large, steele2013validation}).  



 In the 1980's, electrophoretically
detectable genetic variation, in the form of allozymes
\citep{ayala1972allozymes,allendorf1981use}, was used to
establish a program of GSI for Chinook salmon,
{\em Oncorhynchus tshawytscha} \citep{milner1982genetic}.  Extensive sampling
revealed that these allozyme markers
exhibited different allele frequencies among major stocks of Chinook salmon on the West Coast.
These allele frequency differences make GSI possible, and
\citet{milner1985genetic} soon showed that proportions of stocks in
the Washington state coastal troll fishery could be estimated by GSI.
Since that time, with the development of novel molecular markers, and, now
with increasing capacity to sequence genomic material, the scope and scale of GSI
has expanded considerably.

By using greater numbers of more variable markers than the allozymes available
in the 1980s it is possible to accurately
identify the population of origin of individual fish, rather than simply estimating
aggregated stock proportions.  It is also possible to resolve populations of fish that
are much more closely related than before.  Furthermore, reference data sets with genotypes
from hundreds of populations throughout the range of multiple species of salmon and
other anadromous species
\citep{seeb2007development,gilbey2018microsatellite,barclay2019genetic} now exist, and are routinely used to assign fish caught thousands of
kilometers from their natal streams to their stock of origin. Applications include estimating fishery
composition \citep{satterthwaite2015stock}, providing real-time information for genetics-informed fishery closures \citep{beacham2004dna}, assessing the spatial distribution of different stocks in the
ocean \citep{urawa2009stock} and their temporal distribution in upstream migrations
\citep{hess2014monitoring},  and monitoring  bycatch \citep{hasselman2016genetic} or illegal captures \citep{wilmot1999origins} in marine fisheries.

Although sequencing costs continue to decline, they are high enough
that there remains a tradeoff
between reference baselines that include information from a large number of
populations across a broad scale, and those that have been tailored
to distinguish between closely related populations on a smaller, regional scale.
Because of cost considerations, reference baselines that include populations across a broad 
spatial scale may include only a few populations from each sub-region.  Furthermore,
baselines tailored to a specific region often assemble markers that specifically
show allele frequency differences between the closely related---and hence difficult
to resolve---stocks within the region.  Consequently, regionally targeted baselines typically
outperform broad-scale baselines in resolving populations within the region.

The ascendance over the last decade of parentage based tagging (PBT) as a management
tool for Pacific salmon adds another factor to consider when developing a GSI baseline.
Since the first proposal \citep{anderson2005description} to replace or augment the coded-wire tag
programme \citep{nandor2010overview}
with PBT, it has been noted that one of the major advantages of a genetic program for
PBT is that the genetic markers used for PBT could also be useful for GSI (and
vice versa).  This 

Here, we present a reference baseline for Chinook salmon, targeted to the complex population 
structure within the Central Valley of California. Chinook salmon of the two main river basins---the 
Sacramento and the San Joaquin---within the Central Valley exhibit the greatest run-timing diversity
within the species.  With four recognized ecotypes, delineated primarily on the basis of run timing
(Fall-, Late-Fall-, Winter-, and Spring-run), adult Chinook salmon can be found migrating or residing
in freshwater every month of the year in California. NEED MORE HERE. 


 \section*{Methods}

More verbiage

\subsection*{Population Sampling}

We sampled fish for the reference baselines from 7 locations on the across the two
main tributaries (the Sacramento River and the San Joaquin River) within
California's Central Valley. The baseline also includes samples collected from
one location in the Klamath River, and three rivers of the California coast
Fig.~\ref{fig:map}
%%%%%%%%%%%%%
\begin{figure}
\newcommand{\mapcap}{\footnotesize  A map of sampling locations represented in the reference 
baseline.  The colored circles aside
each location name represent the different run-timing groups of Chinook salmon
included the samples from the location.  Color codes are as follows: \Wball~=~Winter run,
\LFball{}~=~Late-fall run, \Fball~=~Fall run, \Sball~=~Spring run.}
\begin{center}
\includegraphics[width=\columnwidth]{images/map-crop.pdf}
\end{center}
\caption[\mapcap]{\mapcap}
\label{fig:map}
\end{figure}
%%%%%%%%%%%%%%%

We are going to stick a table in here:
\begin{table}
\caption{Gonna make a caption. UKTR: Upper Klamath and Trinity Rivers ESU;
CVFR: Central Valley Fall Run ESU; CVSR: Central Valley Spring Run ESU; CVWR:
Central Valley Winter Run ESU
}
\end{table}


\subsection*{Genetic Variation}

\subsection*{Power for genetic stock identification and population assignment}

\subsection*{Power for relationship inference}

\begin{figure*}
\newcommand{\examplecap}{\footnotesize This is just an example figure thrown
in here for now to illustrate the formatting and how the captions are put in. It is not what
we will use for the paper, but we will make another Joseph and the Magic Technicolor
assignment table for this paper.  {\bf a)} Pairwise $F_\mathrm{ST}$ values.
{\bf b)} Assignment rates.}
\begin{center}
\includegraphics[width=0.35\textwidth]{images/assign_and_fst_table.pdf}
\end{center}
\caption[\examplecap]{\examplecap}
\label{fig:example}
\end{figure*}


\section*{Discussion}

When, and if, sequencing costs drop sufficiently, it may  be possible to include
all different populations of a single species within a single standardized reference baseline
that performs equally well at broad and regional scales. A statistical approach for population
assignment from low coverage whole genome sequences appeared recently
\citep{desaixINPRESSpopulation}, and its authors noted that such a data type could
be ideal for producing reference baselines simultaneously applicable to broad scale
coverage and high resolution within sub-regions.   For the present, however,
baselines tailored to specific regions are essential for regional management questions.
It should be noted that recent 

