%!TEX root = main.tex

\section*{Introduction}

Genetic markers have long been used in the
management of Pacific salmon, and salmon conservation and management have been
at the forefront of a number of advances in molecular ecology,
from data generation techniques \citep{clemento2011discovery,campbell2015genotyping,mckinney2017managing,baetscher2018microhaplotypes}
to statistical methodology
\citep{smouse1990genetic,anderson2002model,pella2006gibbs,anderson2010assessing}.
Two broadly applicable techniques that have been
actively fostered by the Pacific salmon research community are genetic
stock identification
(GSI:~\citealt{milner1982genetic,beacham2004dna,seeb2007development})
and parentage based tagging
(PBT:~\citealt{anderson2006power, garza2007large, abadia2013large, steele2013validation}).



 In the 1980s, electrophoretically
detectable genetic variation, in the form of allozymes
\citep{ayala1972allozymes,allendorf1981use}, was used to
establish a program of GSI for Chinook salmon,
{\em Oncorhynchus tshawytscha} \citep{milner1982genetic}.  Extensive sampling
revealed that these allozyme markers
exhibited different allele frequencies among major lineages, or stocks, of Chinook salmon on the West Coast.
These allele frequency differences make GSI possible, and
\citet{milner1985genetic} soon showed that proportions of stocks in
the Washington state coastal troll fishery could be estimated by GSI\@.
Since that time, with the development of novel molecular markers, and, now
with increasing capacity to sequence genomic material, the scope and scale of GSI
has expanded considerably.

By using greater numbers of more variable markers than the allozymes available
in the 1980s it is possible to accurately
identify the population of origin of individual fish, rather than simply estimating
aggregated stock proportions.  It is also possible to resolve populations of fish that
are much more closely related than before.  Furthermore, reference data sets with genotypes
from hundreds of populations throughout the range of multiple species of salmon and
other anadromous species
\citep{seeb2007development,gilbey2018microsatellite,barclay2019genetic} now exist, and are routinely used to assign fish caught thousands of
kilometers from their natal streams to their stock of origin. Applications include estimating fishery
composition \citep{satterthwaite2015stock}, providing real-time information for genetics-informed fishery closures \citep{beacham2004dna}, assessing the spatial distribution of different stocks in the
ocean \citep{urawa2009stock} and their temporal distribution in upstream migrations
\citep{hess2014monitoring},  and monitoring  bycatch \citep{hasselman2016genetic} or illegal captures \citep{wilmot1999origins} in marine fisheries.

Although sequencing costs continue to decline, they are high enough
that there remains a tradeoff
between reference baselines that include information from a large number of
populations across a broad scale, and those that have been tailored
to distinguish between closely related populations on a smaller, regional scale.
Because of cost considerations, reference baselines that include populations across a broad
spatial scale may include only a few populations from each sub-region.  Furthermore,
baselines tailored to a specific region often assemble markers that
show allele frequency differences between the closely related---and hence difficult
to resolve---stocks within the region.  Consequently, regionally targeted baselines typically
outperform broad-scale baselines in resolving populations within the region.

Over the last decade an additional genetic method, parentage based tagging, or PBT,
\citep{anderson2005description,steele2019parentage}
has become established as an extremely valuable management
tool for Pacific salmon.  The availability of such family-based methods adds another factor to
consider when developing a GSI baseline.
Since the first proposal \citep{anderson2005description} to replace or augment the coded-wire tag
programme \citep{nandor2010overview}
with PBT, it has been noted that one of the major advantages of a genetic program for
PBT is that the genetic markers used for PBT could also be useful for GSI (and
vice versa).
Thus, any panel of markers to be used for GSI (or PBT) should also be evaluated on its
utility for PBT (or GSI).

PBT has been remarkably successful in fisheries
management, having been used for over a decade in the management of Chinook salmon
and steelhead trout in major basins of the Columbia River \citep{steele2019parentage,horn2023multigeneration}, and having been employed to dramatically further our
understanding of the genetic inheritance of key traits in salmonids
\citep{abadia2013large,beulke2023distinct}. However, PBT is just one subset of a whole
family of statistical genetic methods employing relationship inference to learn about
populations.  For example, inference of the full siblings amongst a sample of fish
provides information about the effective number of adults producing offspring
\citep{waples2011inbreeding,wang2023estimating}, which can
be a valuable source of information when sampling of the adults is not possible.
Accordingly, marker panels should also be evaluated on their
capacity to resolve full-sibling relationships.


Here, we present a reference baseline for Chinook salmon, focusing on the regional scale of
rivers within the state of California, and particularly targeted to the complex population
structure of Chinook salmon within California's Central Valley (CCV). Chinook salmon of the two main CCV river basins---the
Sacramento and the San Joaquin---exhibit the greatest run-timing diversity
within the species.  With four recognized ecotypes, delineated primarily on the basis of run timing
(fall-, late-fall-, winter-, and spring-run), adult Chinook salmon can be found migrating or residing
in freshwater every month of the year in California \citep{fisher1994past}.

Chinook salmon are the largest of the Pacific salmonids and have historically been the target of extremely high-value and culturally important
fisheries \citep{myers1998status}. Moreover, because of their high degree of ecotypic variation, they have provided fishery opportunities in many
different seasons and geographic locations \citep{healey1991life}. However, recent declines in population sizes, from the Yukon River in the Arctic north,
to the southern extent of their range in California, has led to multiple fishery closures to protect
less productive stocks \citep{lindley2009caused}. The co-occurence of fish from relatively productive and relict populations reaches its paragon in California,
with the largest remaining ocean fisheries for Chinook salmon targeting the Central Valley fall-run stock that spawns in different tributaries of the
same river basin as the highly endangered and phenotypically distinct winter-run stock and the threatened spring-run stocks \citep{satterthwaite2015stock}.

In spite of their distinct phenotypes and life history patterns, the four ecotypes of Chinook salmon in the Central Valley are closely related
\citep{clemento2014evaluation} and share recent common ancestry that is independent of ongoing migration between these lineages. As such, GSI
has been particularly challenging in the CCV, with at least one of the distinct ecotypes, the late-fall-run, unresolvable with previous GSI marker sets
and with unsatisfactory power for discriminating the protected spring-run stocks from the
harvested fall-run stock \citep{seeb2007development,clemento2014evaluation}.

In the following, we present a new reference baseline that includes 1,636 Chinook salmon individuals
from 17 collections, genotyped at 204 microhaplotypes \citep{baetscher2018microhaplotypes} from throughout the genome.
This baseline is
highly effective for GSI within California and the markers within it provide ample power
for PBT, thus enabling a highly effective and efficient integrated GSI/PBT monitoring and evaluation program 
\citep{garza2007large,beacham2021parentage}.
We describe how we identified and compiled the markers, including several new markers
that are particularly effective at distinguishing between closely related groups of fish,
such as late-fall and fall-run Chinook salmon in the CCV.  We also provide a comprehensive analysis of this
set of markers for inference of parents and siblings within different groups of populations.



 \section*{Methods}

\subsection*{Population Sampling}

We compiled samples of fish for the reference baselines from 13 locations within California, and one
within Oregon. This set of populations and stocks includes all of the previously described lineages of Chinook salmon in the southern portion of the range and most of the populations that are known to have significant genetic differentiation in this region.
These locations included sites in the two
main tributaries (the Sacramento River and the San Joaquin River) within
California's Central Valley, and the two main tributaries (Klamath and Trinity Rivers) in the Klamath basin in Northern California, and
three rivers of the California coast (Russian, Eel, and Smith)
(Fig.~\ref{fig:map}, Table~\ref{tab:samples}).
%%%%%%%%%%%%%
\begin{figure}
\newcommand{\mapcap}{\footnotesize  A map of sampling locations represented in the reference
baseline.  The colored circles aside
each location name represent the different run-timing groups of Chinook salmon
included in the collections from the location.  Color codes are as follows: \Wball~=~Winter run,
\LFball{}~=~Late-fall run, \Fball~=~Fall run, \Sball~=~Spring run.
Names of each location are followed by the location codes (Tab.~\ref{tab:samples}).
Abbreviations used in location names as follows:
R. = River; Ck. = Creek; H. = Hatchery; NFH = National Fish Hatchery.}
\begin{center}
\includegraphics[width=\columnwidth]{images/map-crop.pdf}
\end{center}
\caption[\mapcap]{\mapcap}
\label{fig:map}
\end{figure}
%%%%%%%%%%%%%%%

Collections in each location were separated into fish from the different adult migration timing ecotypes according
to a variety of criteria.  Winter-run Chinook salmon are propagated at the Livingston
Stone National Fish Hatchery for supplementation (released as juveniles into the river) and as a captive broodstock;
sampling of this population was performed by hatchery staff during spawning. In Central Valley Rivers with both
spring-run and fall-run ecotypes (i.e. Mill, Deer and Butte Creeks), previous research has documented different  (but overlapping) ranges of spawning time (Julian date) for each ecotype \citep{fry1961king,yoshiyama1998historical}.
As such, sampling targeted carcasses found in each of these
rivers at times representative of their run-timing designation, which we were able to further confirm using their genotypes
within the region of strongest association (RoSA) on chromosome 28 \citep{thompson2020complex}.
In the San Joaquin River, in 2016, at the time of sampling, only
fall-run Chinook salmon adults were found, so the samples there were
easily categorized. Run-timing
designation at the Feather River Hatchery is more complicated but it is managed by the hatchery practices in place:
early returning fish in May and April (expressing the spring-run life history) are visibly tagged and re-released
to the river. Subsequently in the Fall, when hatchery raceways are reopened, only fish with the visible tag are included
as spring-run broodstock, and the rest are included in the fall-run broodstock. We sampled putative spring-run and fall-run fish 
from these two broodstocks. Similar to broodstock collections in the CCV, samples from spring- and fall-run fish in the Trinity River were selected on the basis of spawn timing at the Trinity River Hatchery with migration-timing ecotype then confirmed by RoSA genotype. Elsewhere, samples were taken from live broodstock at Cole Rivers (Rogue), Rowdy Creek (Smith), Iron Gate (Klamath) and Warm Springs (Russian River) Hatcheries, where only fall-run fish have been documented. Samples from the Eel River were taken from live fish ascending a ladder in the upper basin and were confirmed as fall-run ecotype by RoSA genotype. 



In total, the samples in the baseline represent fish from five (Table~\ref{tab:samples}) different Evolutionarily Significant
Units (ESUs) of Chinook salmon \citep{waples1991pacific}.
%%%%%%%%%%%%%%%%
\begin{sidewaystable*}
\caption{\footnotesize Collections of fish in the reference baseline.  The Sampling Months column shows
a pictorial representation of the proportion of each collection sampled in each month.  Colors in
this column correspond to run-timing groups as described in the caption of Fig.~\protect\ref{fig:map}.}
\label{tab:samples}
{\small
\begin{tabular*}{\linewidth}{@{\extracolsep{\fill}} lllllllrcr}
\hline\hline
\vspace*{0.4ex}
\input{inputs/samples-table.tex}
\end{tabular*}
}
\rule{3cm}{0.3pt}

{\scriptsize
$^1$R. = River.\\
$^2$Ck. = Creek; NFH = National Fish Hatchery; R. = River; H. = Hatchery.\\
$^3$Reporting Unit = an aggregation of the collections into groups relevant to management and
which are accurately resoluble with the genetic data in the baseline.
$^4$ESU = Evolutionarily Significant Unit; CVFR = Central Valley Fall Run ESU; CVSR = Central Valley Spring Run ESU; CVWR =
Central Valley Winter Run ESU; CC = California Coastal ESU; UKTR = Upper Klamath and Trinity Rivers ESU. \\
$^5N_\mathrm{day}$: the number of samples for which the specific sampling day was recorded.
}
\hrule\vspace*{0.3ex}\hrule
\end{sidewaystable*}
%%%%%%%%%%%%%%%%%%%%%

\subsection*{Genetic Markers}

The genetic markers in the baseline include newly discovered microhaplotypes, SNP assays
that were translated into amplicons, markers associated with phenotypes, as well
as species- and sex-specific markers.
These genetic markers were compiled from several different
efforts, as detailed in the sections below.

\subsubsection*{Novel microhaplotype discovery}

We identified candidate loci with multiple single-nucleotide polymorphisms (SNPs) among multiple
populations using reduced-representation genome sequencing of 2--4 Chinook salmon, each, from a number of
populations ranging from California to British
Columbia and the interior Columbia River (Table~\ref{tab:ascert-panel}).
Genomic DNA was extracted using Qiagen DNeasy 96 Blood and Tissue Kits on a BioRobot 3000
(Qiagen, Inc). We used double-digest restriction-site associated DNA sequencing (ddRAD) with different size selections (400 bp and 500 bp
fragments) to produce a broad array of locus targets. Library preparation and sequencing methods followed those of
\citet{peterson2012double} with the modifications described in \citep{baetscher2018microhaplotypes}.
Sequencing was performed on a MiSeq (Illumina, Inc) using $2 \times 300$-cycle paired-end sequencing.

Stacks v1.45 \cite{catchen2013stacks} was used to analyze the ddRAD sequence data.  As there was not a well-assembled Chinook salmon genome
at the time, a {\em de novo} assembly was performed.  Initial filtering retained RAD loci with two or more SNPs present
in at least 10 samples. After this initial filtering, there were 3,931 RAD loci retained in the ddRAD data with 400 bp size selection and
4,794 retained in the data with 500 bp size selection.
Gene regions were then selected that (1) contained multiple SNPs within a 100 bp window, and (2) showed haplotype variation amongst the winter run and at least one other population.
A 100 bp window was used because the success of multiplex PCR and MiSeq sequencing is diminished in amplicons larger than 100 bp. Primer3
software \citep{untergasser2012primer3} was used to design primers for 229 gene regions that met these criteria.

These gene regions were PCR-amplified and sequenced on a MiSeq sequencer using a $2\times 75$-cycle kit.
The sequences from each locus were mapped to the de novo reference sequences (produced by Stacks) with
the Burrows-Wheeler aligner using the mem algorithm \citep{bwa-mem2009}. 
Subsequently, alignmets were sorted with SAMtools \citep{li2009sequence} and provided to
FreeBayes \citep{garrison2012haplotype} for variant calling.
FreeBayes output was filtered to remove indels and retain variants with a Phred-scaled quality score of 30
or greater and a read depth of 10 or higher. Further filtering and analysis of the loci, including
assessment of Hardy-Weinberg equilibrium \citep{hardy1908mendelian} was conducted in MICROHAPLOT (Ng et al., DOI: 10.5281/zenodo.820110).  Loci that produced more than two haplotypes within an individual, as well as those clearly violating Hardy-Weinberg equilibrium within populations were removed.


\subsubsection*{Conversion of existing SNPtype assays}

A number of SNP markers from an earlier discovery effort \citep{clemento2011discovery} had already
been validated as useful for GSI
and PBT in California \citep{clemento2014evaluation}. These markers were originally assayed
using TaqMan (Life Technologies Corp., Carlsbad, CA) probes or SNPtype assays (Standard BioTools Inc., South San Francisco, CA).
In order to type these markers using amplicon sequencing, we followed the methods in
\citet{campbell2015genotyping} and added the Illumina small RNA sequencing primer and
read 2 sequencing primer to the SNPtype Specific Target Amplification Primer and Locus
Specific Primers respectively.  We then dropped any loci that produced genotypes that were
frequently inconsistent with genotypes produced with the SNPtype assays and any loci that
did not amplify well or those that amplified so well that they were grossly over-represented
among the reads on the sequencer.  We
used freebayes \citep{garrison2012haplotype} to identify all variants on the fragment (not just the SNPs targeted by the original TaqMan assays).  This allows us to score each locus as a
microhaplotype \citep{baetscher2018microhaplotypes} when multiple variants occur in the sequence.  We filtered out variants
that were considered multinucleotide or complex variants by freebayes. The resulting genetic markers are hereafter referred to as 'SNPlicons'.


\subsubsection*{Identifying the Late-Fall Associated Region (LFAR) and designing markers therein}

To identify loci with large allele frequency differences between late-fall and fall-run Chinook salmon,
we first conducted a simple case-control GWAS using the low-coverage whole-genome sequencing
data of \citet{thompson2020complex}, mapped to Otsh\_v1.0.  We conducted
two separate association-study comparisons.
The first involved the 16 late-fall-run fish as cases with 16 fall-run fish from the Feather River Hatchery as controls.
The second involved the same 16 late-fall fish as cases, but used 16 fall-run fish from the San
Joaquin River as controls.  The association studies were performed for each chromosome with
ANGSD \citep{pmid21663684,korneliussen_angsd_2014}, using the following command: {\footnotesize\tt angsd -yBin \{ybin\}  -r \{chrom\}
-minMapQ 30 -minQ 20 -minInd 12 -doAsso 1 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP\_pval 1e-6
-out \{prefix\}  -bam \{bamlist\}}, where {\tt \{ybin\} } is the path to a file indicating the cases and
controls,
{\tt \{chrom\} } indicates the chromosome name, {\tt \{prefix\} } is the prefix to use for output files, and
{\tt \{bamlist\} } is the path to the text file holding paths to the binary alignment map (BAM) files
of aligned sequence data, one for each individual, ordered as in {\tt \{ybin\}}.

There was only one peak in association scores that was shared between both comparisons (see
Results). From this peak we identified candidates for follow-up genotyping by including the 10
SNPs with the lowest association $p$-values, and supplementing those with additional SNPs that
showed:
\begin{itemize}
\item An estimated absolute allele frequency difference between late-fall and all 32 fall-run fish, $|d| >0.75$,  with an approximate lower confidence interval of $|d|$ greater than 0.5,  or
\item $|d| >0.5$ with a lower confidence interval $>$0.25 and
with either fall-run or late-fall run being nearly fixed for one of the alleles, or
\item An annotation from snpEff \citep{cingolani2012program} of “High” or “Moderate”, or
\item $|d|>0.48$ and a lower confidence interval of $>$0.25 and a
genomic  coordinate $> 2.2$~Mb.
\end{itemize}
The final condition was implemented to gather several SNPs within the flanking region to the right of
the main peak of association.

For the candidate markers found by the above process, we designed amplification primers using
Primer3 \citep{koressaar2007enhancements,untergasser2012primer3} and tested amplification in a
subset of Chinook salmon
samples. The primer pairs that amplified sequence that reliably mapped back to the association-peak region were then typed
in a separate collection of 1,638 Sacramento River Chinook salmon sampled from the fish trap below Keswick Dam, including 1,169 winter run, 281 spring
run, and 188 fall/late-fall run. We divided the fish identified as fall/late-fall (via the 125-locus genetic
stock identification data set of \citealt{thompson2020complex} into fall run and late-fall run according
to their sampling date at Keswick:  99 fish arriving before Dec 26 were classified as fall run and 89 fish arriving after Dec 26 were
classified as late-fall run.   The allele frequencies of the target SNPs at the remaining amplicons
were
estimated for each of these four groups: winter, spring, fall, and late-fall, and we retained
amplicons showing large allele frequency differences between late-fall fish and all other runs.


\subsubsection*{Winter-run associated polymorphisms (WRAPs)}

Using the whole genome sequencing data of \citet{thompson2020complex}, we also sought
genomic regions that might provide diagnostic markers for the the Sacramento winter-run
Chinook salmon.  This ecotype is highly differentiated from all other Chinook salmon populations
throughout its genome, which complicates the process of finding diverged genome regions using
a simple association study, with only 16 samples, as done to identify the LFAR.  Consequently we pursued a bespoke
analysis to find regions we could expect to have large allele frequency differences between
the winter-run and all other ecotypes of Chinook salmon in California's Central Valley.   This analysis
screened for regions with large allele frequency differences across large blocks of the genome, and is
described in the supplemental text section \ref{sec:wrap-methods}.

\subsubsection*{Run-timing associated markers}

We include six amplicons that capture the eight  SNPs within the ``Region of Strongest Assocation'' (RoSA)
between spring-run and fall-run fish on Chromosome 28 listed in Table~S3 of \citet{thompson2020complex}.
We add three more amplicons in the same RoSA region to the panel, as well as one more amplicon that
allows genotyping of snp670329, which was discovered and described by \citet{thompson2019anthropogenic}.
Including all these markers in the panel allows them to be routinely typed, creating
 a data base for understanding the spatial and temporal
distribution of the alleles in this genomic region; however we have omitted these markers for calculating likelihoods for
GSI, since their distribution across populations in the Feather River has been clearly and evidently
influenced by hatchery practices (see Results), it is not necessary to include them for accurate assignment,
and because the 12 SNPs within these three amplicons are typically in profound linkage disequilibrium (LD)
making them inappropriate for use as separate markers in most software for population assignment or GSI, that
typically assumes markers are not in LD within the component populations of the baseline.




\subsubsection*{Additional phenotypically associated markers}

The baseline also includes three amplicons with genetic variation
in two genes, VGLL3 and SIX6,
that have been found to be associated with phenotypes related to reproductive maturity in both
Atlantic \citep{barson2015sex} and Pacific \citep{waters2021heterogeneous} salmonids. We include
them both to provide additional variation for population genetic analyses and to
monitor populations for potential phenotypic patterns.

We include a marker that targets the sex-determining region ({\em sdY} on the Y chromosome in Chinook salmon.
This marker characterizes genetic sex in Central Valley populations \comment{\citep{von2015development}}.
And, finally, we include marker Ots\_coho001\_05\_32691399 which is fixed for alternate alleles in
coho salmon and Chinook salmon.  This is helpful when samplers have unwitting
collected coho salmon from ocean catches.

\subsection*{Localizing markers within the genome}

The novel microhaplotype markers and the SNPlicon targets were
developed prior to the advent of a well-assembled genome for Chinook salmon.
As a consequence, many of these markers had
been used previously without certainty about their location in the genome.
After designing and validating an amplicon-sequencing approach to type both of these
groups of markers we identified their locations within the genome by mapping the
consensus sequences (roughly 100--300 bp) around the markers
to both the Chinook salmon version 1 (Otsh\_v1.0; \url{https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_002872995.1/}) and version 2  (Otsh\_v2.0; \url{https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_018296145.1/}) genome assemblies,
using both bwa-mem \citep{bwa-mem2009} and the BLAST-like Alignment tool \citep{kent2002blat}.
A similar mapping exercise was performed for the remaining markers so that we can
report their locations in both Otsh\_v1.0 and Otsh\_v2.0.


\subsection*{Amplicon sequencing, microhaplotype calling, and further quality control}

Amplicon sequencing was performed with the Genotyping-in-Thousands by sequencing (GT-seq)
method \citep{campbell2015genotyping}. All loci were
multiplexed with primer concentrations ranging from 0.083 uM to 0.5 uM to increase
uniformity of read depths across all loci. Sample normalization was
performed using the SequalPrep Normalization Plate Kit (Applied Biosystems) according
to the manufacturer's instructions. Samples were pooled by plate
and then purified using Agencourt AMPure XP magnetic beads.
Libraries were quantified by qPCR with the NEBNext Library Quant Kit for Illumina (New
England BioLabs) before dilution and pooling for sequencing.
All samples were sequenced on a MiSeq using $2\times 75$-cycle paired-end sequencing protocols.
Primer sequences and concentrations for making the primer pool are in Supplemental Table 1.

For evaluation of candidate loci, dual-barcoded sequences were used
to identify individual tissue samples using the MiSeq analysis software (Illumina).
The paired-end sequence reads were joined together using Fast Length Adjustment
of SHort reads (FLASH: \citealt{magovc2011flash}) and mapped to an indexed sequence reference created from the Stacks target loci using BWA (Li and Durbin 2010). Mapped reads were converted to BAM files using SAMtools (Li et al. 2009) and freebayes (Garrison and Marth 2012) was used to call variants.

For genotyping of baseline samples, final filtering at each locus required a read depth of 20 and a read depth ratio of 0.30 (i.e., for heterozygotes the lower read depth haplotype must have 30\% of the total reads at that locus). MICROHAPLOT was used to call alleles and genotype processing was performed with R scripts.

Each locus was assessed for the presence of null alleles using the R package `whoa'
(\url{https://github.com/eriqande/whoa}; {\em see} \citealt{hendricks2018recent}).

\subsection*{Population genetic analyses}

Basic data summaries were calculated for each population/collection
in the reference baseline
using simple operations in the R programming language, Version 4.3
\citep{rcore}. We started with a simple tally of the number of alleles
found at each locus.  To provide a summary of the genetic variation and
missing data we further summarized:
\begin {itemize}
\item the average over loci of the number of fish
in which each locus was successfully genotyped, $\bar{N}$;
\item the average number of alleles per locus in each collection with all collections
downsampled to the smallest number of genotyped individuals at each
locus, $\bar{N}_\mathrm{A,ss}$;
\item the fraction of polymorphic
loci in each collection after subsampling to the smallest number of genotyped
fish, $\bar{P}_\mathrm{poly,ss}$;
\item the expected heterozygosity as the average over loci
of one minus the frequencies of homozygotes at each locus expected from the
estimated allele frequencies, $\bar{H}_\mathrm{exp}$; and
\item the observed heterozygosity as the average over
loci of the fraction of heterozygous genotypes, $\bar{H}_\mathrm{obs}$.
\end{itemize}
When subsampling each collection to the minimum number of genotyped
fish across all collections at a locus, we subsampled individuals without
replacement, taking the average of 1,000 different random subsamples.

We calculated \citet{weir1984estimating}'s pairwise $F_\mathrm{ST}$ between
all collections, using the {\footnotesize\tt pp.fst} function from the R
package `hierfstat' \citep{hierfstat}. Finally, we evaluated population
structure in the data by performing unsupervised clustering 
using the program STRUCTURE \citep{pritchard2000inference,falush2003inference}.
For this analysis, we omitted the run-timing associated markers.
We performed 20 replicate runs at each value of the assumed number of
subpopulations, $K$, in $\{2, 3, 4, 5, 6, 7\}$
using default settings of the program.  STRUCTURE output was
a summarized and visualized using CLUMPAK
\citep{kopelman2015clumpak}.






\subsection*{Power for genetic stock identification and population assignment}

To assess the power of the marker panel for GSI, we used the R package
`rubias' \citep{moran2019bayesian}.  We used the function {\tt self\_assign()}
to perform assignment of each individual in the reference baseline to one of the
collections within the baseline.  Each fish was removed from the reference baseline
when calculating the likelihood that it originated from the collection it actually belongs
to, using a classical leave-one-out approach to eliminate the inflated estimates of
power that might occur without such a procedure \citep{anderson2008improved}.
The scaled likelihoods of collection membership returned by
{\tt self\_assign()} were summed within reporting units, and fish were
assigned to the reporting unit with the largest value of that sum.  Subsequently,
fish were assigned to the collection within that reporting unit with the highest
scaled likelihood.  We considered two thresholds for assignment.  In the first,
all fish are assigned, in the second only fish with a summed (i.e., summed
over all collections in a reporting unit)
scaled likelihood greater than 0.8 were assigned.

Finally, we investigated the matrix of assignments and misassignments
broken down according to the genotype of the fish at the markers within
the RoSA.  For this we identified the RoSA genotype (EE, EL, or LL, see
\citealt{thompson2020complex}) from the concordance of the genotypes
at all 8 markers described in \citep{thompson2020complex}, requiring no missing
data at any of those 8 markers, and discarding data from 12 individuals
in which the E or L haplotypes had clearly recombined.



\subsection*{Power for relationship inference}

To explore the utility of the marker panel for relationship inference, we used
the R package `CKMRsim'
(\url{https://github.com/eriqande/CKMRsim})
to estimate the false-positive rates (FPRs) expected at
different values of the false negative rate (FNR) for a variety of different pairwise
relationships in different populations.  For these purposes, we divided the
collections into 10 different groups.  The members of each group are
relatively, genetically similar, but the groups do not necessarily correspond to reporting
units, because, in some cases, we desired FPRs and FNRs for a single
collection.   For all cases, we assumed that the genotyping error rate
was 1\% per locus and the genotyping error model was the default
"True-genotype-independent" model.


For errors involving misidentifying
unrelated (U) pairs as parent-offspring (PO), full-sibling (FS), or half-sibling (HS)
pairs we used importance sampling to estimate the very small FPRs.  The
FNRs were estimated in each case using simple Monte Carlo.  For the FS and
HS cases the FNRs were calculated from Monte Carlo sampling while taking account
of the physical linkage of the markers by inputting their genomic positions into
`CKMRsim' and using the package's interface to the MENDEL \citep{lange2013mendel}
software to simulate genotypes of related pairs
in the presence of physical linkage.  For PO pairs (and U pairs),
FNRs were calculated without reference to the genomic locations of the markers
because physical linkage does not
affect the distribution of log-likelihood ratios in those two relationship types,
because there is no variation, anywhere
in the genome, in the number of gene copies shared identical by descent between
a parent and its child or between two unrelated individuals.

For errors involving misidentifications between the relationship
types of avuncular (AN: aunt-niece, uncle-nephew, etc.), PO, FS, and HS,
we used regular Monte Carlo to estimate the FPRs and FNRs, because
importance sampling cannot be used in those cases. Since there are
expected to be far fewer such relationships than the number of unrelated
pairs, the relevant FPRs are large enough that they can be reliably estimated
(at least down to rates of $10^{-3}$) without importance sampling.
For AN, FS, and HS, both the
FPRs and FNRs were estimated while taking account of physical linkage
of the markers, as described above.

\section*{Results}

\subsection*{Genetic Markers}

A summary, which includes genomic location, consensus sequence, and primer sequences,
of all the amplicons used in the baseline is available
in Data Supplement~1.

\subsubsection*{Additional microhaplotype discovery}

We tested the 229 candidates from our novel microhaplotype discovery process and
initially retained the 125 markers described in \citep{thompson2020complex}
for use in stock identification within
the Klamath Basin.  However, further evaluation of those markers in other
populations led us to remove some of the markers.
Twelve loci appeared to have null alleles in the winter-run population and were removed.
Additionally, two loci were monomorphic in the winter-run, even though they contained
multiple alleles in other populations, and an additional 5 amplicons were deemed
difficult to score in routine application of the baseline, and were removed, leaving 106
novel microhaplotype amplicons in the current panel.


\subsubsection*{Conversion of existing SNPtype assays}

Of the 96 existing SNPtype assays, we were able to convert 78
into reliable amplicon-sequenced assays that were included in the panel.



\subsubsection*{Late-fall Associated Region Markers}

The case-control association test revealed a single peak on chromosome 34 with large allele frequency differences
between Central Valley late-fall and fall-run fish.  This peak was evident both in the comparison of
late-fall fish to fall-run fish in the Feather River Hatchery and of late-fall fish to the fall-run fish in
the San Joaquin River (Fig.~\ref{fig:lfar-assoc}).




The Manhattan plot of Fig.~\ref{fig:lfar-assoc} shows sporadic SNPs with low association
$p$-values.  These are not shared between the two different late-fall to fall comparisons, and
sites that we investigated were concentrated in areas of poor mapping, suggesting that their
$p$-values were artifactual.  By contrast, a large number of sites near the prominent peak on
chromosome 34 had large allele frequency differences between late-fall and fall run.  Taking the
10 SNPs with the lowest association $p$-values, and including additional ones from the filtering criteria given in the
Methods, yielded 49 candidate markers for which we designed primers for amplication.  Only 8
of these 49 primer pairs yielded reliable amplification and mapping.   Of these 8, two of the SNPs
showed inconsistent genotyping and were removed from consideration.  Of the remaining 6, only 3
showed marked allele frequency differences ($> 0.7$) between late-fall and fall run.  These three
SNPs (located at: Chr34:828,768,  Chr34:865,057, and Chr34:1,063,084 in the Otsh\_v1.0 assembly)
also had pronounced differences in allele frequency between
late-fall and spring or winter run.

Analysis of the mapping of amplicons for Chr34:828,768
showed that many ($\approx$89\%) of the reads were off-target (aligning to other chromosomes,
etc.), making it costly (in terms of sequencing effort) to include the marker in the baseline panel.
Consequently it was dropped from the panel.  The two remaining loci, which are at coordinates
Chr34:865,057, and Chr34:1,063,084 in the Otsh\_v1.0 assembly are at coordinates
Chr34:954,054 and  Chr34:1,151,868 in the Otsh\_v2.0 assembly, and
the frequency of the different alleles at the two remaining loci
within different reporting units in the baseline are shown in Table~\ref{tab:lfar-freqs}.
%%%%%%%%%%%%%%
\begin{table}
\caption{\footnotesize Allele frequencies across reporting units of the two late-fall associated
markers.  Frequencies are given for the alleles most common amongst late-fall Chinook: nucleotide
{\tt G} at locus Chr34:954,054 and {\tt A} at Chr34:1,151,868.  $N$ is the total number of gene
copies available for estimating the allele frequency. Genome coordinates of SNPs are from Otsh\_v2.0.}
\label{tab:lfar-freqs}
{\footnotesize
\begin{tabular*}{\columnwidth}{@{\extracolsep{\fill}} lrrrr}
\hline\hline
& \multicolumn{2}{c}{\underline{Chr34:954,054}} & \multicolumn{2}{c}{\underline{Chr34:1,151,868}} \\
Reporting Unit & freq. & $N$ & freq. & $N$ \\ \hline
\input{inputs/lfar-freqs.tex}
\end{tabular*}
}
\vspace*{-2.3ex}\hrule\vspace*{0.3ex}\hrule
\end{table}
%%%%%%%%%%%%%%%

\subsubsection*{Winter-run-associated polymorphisms}

Though the winter run fish are easily identified, we undertook an additional effort to
try to find markers that could be diagnostic for the winter run, distinguishing the
winter run from all other groups of Chinook salmon in the CCV.  We were unable to
find a single diagnostic locus; however three markers from that effort are included in
the reference baseline.  The process of discovering them is described in the
supplemental text section \ref{sec:wrap-methods}.

\subsection*{Localizing markers within the genome}

Of the 184 loci that originated from our novel
microhaplotype discovery or from SNPtype assays, 179 of them mapped identically to a single
location on an assembled chromosome in the Otsh\_v2.0 genome, using both bwa mem and BLAT.
Of the remaining 5 loci, one of them was mapped to the same location by BLAT and bwa mem, but
the alignments differed in length, two of them had a single secondary alignment on a different chromosome
that was identical in both bwa mem and BLAT, one of them had only a fragment mapping to
the secondary alignment, and one of them had multiple
non-primary alignments (Supp.~Data~1).

Mapping to Otsh\_v1.0 was similar, except that a greater number of markers mapped to unplaced
scaffolds in Otsh\_v1.0 than in Otsh\_v2.0, likely reflecting the more complete nature of the
second assembly.


\subsection*{Population genetic summaries}

The total number of alleles per locus varied between 1 and 10 (Fig.~\ref{fig:num-alle}), with the
only locus with a single allele being the species-specific marker Ots\_coho001\_05\_32691399
used to distinguish coho from Chinook salmon. Biallelic
loci were most common, at 66 of the total, but two-thirds of the microhaplotype markers
showed more than two alleles, which has been shown to provide significantly more
power for relationship inference than a comparably sized panel of biallelic SNPs
\citep{baetscher2018microhaplotypes}.


The population-genetic summary statistics are presented in Table~\ref{tab:pg-summ}.
%%%%%%%%%%%%%%
\begin{table}
\caption{\footnotesize Simple population genetic summaries. All quantities are means
over loci.  $\bar{N}$ is the number of fish with observed genotypes; $\bar{N}_\mathrm{A,ss}$ is the
average number of alleles and $\bar{P}_\mathrm{poly,ss}$ the fraction of polymorphic loci,
after sub-sampling to the smallest sample size per locus; $\bar{H}_\mathrm{exp}$ and
$\bar{H}_\mathrm{obs}$ are expected and observed heterozygosity, respectively. Population
codes are as given in Table~\ref{tab:samples}}
\label{tab:pg-summ}
{\footnotesize
\begin{tabular*}{\columnwidth}{@{\extracolsep{\fill}} lrrrrr}
\hline\hline
\input{inputs/popgen-summary.tex}
\end{tabular*}
}
\vspace*{-2.3ex}\hrule\vspace*{0.3ex}\hrule
\end{table}
%%%%%%%%%%%%%%%
The average number of fish with non-missing data in each population varied from
a low of 40.2 (TRF) to a high of 279.6 (CHLF).  The average number of alleles
per locus, standardized to the smallest sample size, ranged from 2.32 (SRW) to
2.64 (RRF), with  SRW also having the smallest value of the
standardized fraction of polymorphic markers at 0.88 and RRF sharing the highest value
(0.96) with FHRF and BCkF.
SRW also had the lowest values for expected and observed heterozygosity:
0.338 and 0.329, respectively. While the highest for expected heterozygosity
was 0.401 (SmRF) and for observed heterozygosity, 0.399 (CRHS).

The estimated pairwise $F_\mathrm{ST}$ values between the collections
varied from a low of 0.0 to a high of 0.288 (Figure~\ref{fig:gsi}a).
Notably, most of the largest values
of $F_\mathrm{ST}$ occurred between SRW and another collection.
%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}
\newcommand{\gsicap}{\footnotesize ({\bf a)} Pairwise $F_\mathrm{ST}$ values and allele frequencies,  
and ({\bf b)} self-assignment rates.  The codes of the populations
appear on the outer margins of the table atop cells colored according to the run-timing group
of the collection.  Interior cells are colored according to reporting unit.  In ({\bf a}), for $F_\mathrm{ST}$ values,
the upper triangle was computed using all the genetic markers except for those in the
RoSA on chromosome 28. 
Each cell in the lower triangle shows a small scatter plot of estimated allele frequencies
in the collections---the $x$ value is allele frequencies from the collection in the column and the
$y$ value is from the collection in the row. Frequencies for all alleles are shown, with values within
each cell ranging from 0 to 1.  These frequencies were estimated by, for each locus, randomly
subsampling individuals from each collection to the minimum number of observed alleles in any collection
at the locus.  In ({\bf b}), for self-assignment tallies, the row corresponds
to the true source collection and the column corresponds to the assigned collection.
All individuals were assigned here, using a maximum-scaled-likelihood rule to reporting unit and
then to collection within reporting unit as described in the text. For the assignment matrix resulting
when only fish with a scaled likelihood $>0.80$ to reporting unit are assigned, see Figure~\ref{fig:eighty}}
\begin{center}
\includegraphics[width=\columnwidth]{images/gsi_and_fst_fig-crop.pdf}
\end{center}
\caption[\gsicap]{\gsicap}
\label{fig:gsi}
\end{figure}
%%%%%%%%%%%%%%%%%%%%



At value of $K$ from 2 to 7, clusters identified by the program STRUCTURE
generally corresponded to groupings of related populations, and confirmed
{\em a priori} knowledge about which collections in the baseline could and should be
grouped together into reporting units (Fig.~\ref{fig:struct}).  At every value of $K$, CLUMPAK discerned at least
12 of 20 replicates in the major mode (Fig.~\ref{fig:struct}).   
Clustering solutions in the minor
modes generally converged on a single alternative solution and appear in Fig.~\ref{fig:minor-modes}.
%%%%%%%%%%%%%%%%%%%%
\begin{figure}
\newcommand{\structcap}{\footnotesize Major modes found by STRUCTURE and summarized
by CLUMPAK out of 20 replicates for each value of $K$, the assumed number of
subpopulations. $K$ value is listed at lower right of each barplot, followed by the number
of replicate runs of structure that converged to this major mode.}
\begin{center}
\includegraphics[width=\columnwidth]{images/clumpak-crop.pdf}
\end{center}
\caption[\structcap]{\structcap}
\label{fig:struct}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%
The late-fall run collection (CHLF) appears as a separate cluster at $K$ values
as low as 4, with the inclusion of the LFAR markers.  This is notable, since accurately discriminating between the late-fall and
fall-run Chinook salmon of the Central Valley has been impossible with previously used genetic marker sets.
Another noteworthy feature appears at $K=7$, with STRUCTURE separating the fall-run reporting unit
into two clusters (orange and red) that do not correspond exactly with the individual
collections within the reference baseline. For example, fish from the
orange cluster that predominates in the FRHS collection are also found in the FRHF and MDS collections.


\subsection*{Power for genetic stock identification and population assignment}

Leave-one-out cross-validation (self-assignment) analysis demonstrates that this
reference baseline provides a high degree of accuracy for distinguishing the major
groups of Chinook salmon in California.  The results of the analysis are summarized
in a self-assignment matrix (Figure~\ref{fig:gsi}b).
This figure shows the collections divided into the seven different reporting groups:
SO-NCal-Coast,  Klamath-Trinity, Cent-Cal-Coast,
CV-Winter, CV-Spring, CV-Fall, and  CV-Late-Fall.  These groupings correspond
largely to the divisions in the data found by STRUCTURE, they also correspond
with divisions amongst the stocks as defined for management, and, as seen in the
figure, they correspond with groups of populations that can be reliably distinguished
from one another for population assignment.

Overall, of the 1636 fish in the reference baseline, 1612 (98.5\%) were correctly
assigned to their reporting unit of origin, while 24 (1.5\%) were incorrectly assigned.
Four of those 24 misassignments which involved fish being incorrectly assigned to
SRW, were clearly fish from SRW that had been incorrectly sampled into a different
collection, likely due to straying. (This is evident because SRW are so distinct that it is not possible
to incorrectly identify them).  Of the remaining 20 misassignments,  8 involved fish from the
CHLF (late fall run) collection being misassigned to the CV-Fall reporting unit and 6 were
fish from the MDS (Mill-Deer spring run) collection being assigned to the CV-Fall reporting unit.
It is possible that some of these misassignments represent CV-Fall fish that were incorrectly
sampled as CHLF or MDS; however, unlike with SRW,  we  cannot conclude that
confidently, as there is some degree of overlap in likelihood values between the two.

While fish from CHLF were found to misassign at a rate of 2.7\% (8/300) and MDS at a rate
of 7.1\% (6/84) to CV-Fall, the
converse is not evident---that is, fish from the CV-Fall are misassigned to CHLF at a rate
of only 0.6\% (3/508) and to the CV-Spring reporting unit (to which MDS belongs) at a rate
of only 0.4\% (2/508).

When assignments are only accepted with a scaled likelihood greater than 0.8,
20 assignments are discarded, and half of those (10) were the incorrect assignments.  Thus,
at the 0.8 threshold, the total fraction of correct assignments is 99.1\% (Fig.~\ref{fig:eighty}).
Notably with the $>0.8$ criterion, the misassignment rates from the abundant CV-Fall
reporting unit to CV-Late-Fall or to CV-Spring both drop below 0.2\% (1/502).  

As seen in previous studies, FRHS, the Feather River Hatchery
spring-run stock, is considerably differentiated from the remaining spring-run
stocks in the Central Valley (MDS and BCS). This is reflected in the low
misassignment rates of FRHS fish to the CV-Spring, and is also evident in the
STRUCTURE results.   We note, however, that the RoSA markers provide
the differential signal necessary to distinguish
Feather River spring-migrating fish from the other CCV stocks with a predominantly fall-run
genomic background. This is evident when the assignment matrix is enumerated in
terms of RoSA genotypes (Figure~\ref{fig:rosa-gsi}). In this context, it is clear that the ``fall-run'' hatchery
program in the Feather River both spawns and produces many spring-run fish and RoSA heterozygotes.
Similarly, in other regions, such as the Trinity River, the RoSA markers are the only reliable means to distinguish fall-
and spring-run fish, as they otherwise share genomic background.

In California, only the Sacramento and  Klamath basins have documented spring-run
salmon and their historical occurrence in other basins has been unclear. We find no
instance of spring-run fish in these coastal basins, but did identify one copy of the early-lineage
RoSA haplotype in the Eel River. Moreover, we identified three additional copies of a
recombinant haplotype that carries half of the RoSA SNPs from both the E and L lineages in the Eel and Russian rivers. No other clear recombinants were identified in the study populations, indicating that it arose in the California Coastal Chinook Salmon lineage, and providing further evidence of past presence of RoSA haplotypes associated with early migration.





\subsection*{Power for relationship inference}

The results are summarized in Fig.~\ref{fig:fprs}.
%%%%%%%%%%%%%%%%%%%%%
\begin{figure*}
\newcommand{\fprcap}{\footnotesize False positive rates (FPRs) as a function of false negative
rates (FNRs) for distinguishing between different relationships. Groupings for panels that are neither a previously defined reporting unit nor a single collection (Table \protect\ref{tab:samples}) are as follows: TRH = \{TRS, TRF\}; FRH = \{FRHS, FRHF\}; CVF = \{BCF, MDF, SJRF\}.  {\bf (a)} FPRs calculated
using importance sampling of unrelated pairs of individuals being incorrectly categorized as parent-offspring (PO),
full-sibling (FS) or half-sibling (HS) pairs.  {\bf (b)}  FPRs calculated using regular Monte Carlo
sampling of avuncular pairs (AN: aunt-niece, uncle-nephew, etc.) or full-siblings (FS) or half-siblings (HS) being
incorrectly categorized as parent-offspring (PO); as well as FPRs of parent-offspring (PO) pairs or half-sibling (HS)
pairs being incorrectly inferred as full siblings (FS).  In both {\em a} and {\em b}, 100,000 Monte Carlo replicates
were simulated.  Importance sampling allows estimation of very small false positive rates for unrelated pairs, but
cannot be applied to truly related pairs.  Consequently, it is difficult to accurately estimate small probabilities
($10^{-4}$ or less) in {\bf b}, as indicated by the error bars, which represent the estimate $\pm 2s$ where $s$
is the estimated standard error of the mean of the Monte Carlo sample.  This also is the reason that many of the lines terminate.   }
\begin{center}
\includegraphics[width=\textwidth]{images/fpr-fnr-figure-crop.pdf}
\end{center}
\caption[\fprcap]{\fprcap}
\label{fig:fprs}
\end{figure*}
%%%%%%%%%%%%%%%%%%%%

The `CKMRsim' power analyses for this set of markers demonstrates that they have ample variation
for accurate identification of parent-offspring (PO) and full-sibling (FS) pairs in almost all
realistic situations. The predicted false positive rates from unrelated pairs for both the
parent-offspring (PO) and full-sibling (FS) relationships are
exceedingly low for all population groups, even when using stringent false negative thresholds as
low as 0.05 (Fig.~\ref{fig:fprs}a).
For example, the false positive rate for FS identification is less than one error in 10 million
comparisons of potential siblings in the California Chinook salmon population with the lowest
heterozygosity (SRW) at a false negative rate of 0.05. 

Half-sibling (HS) pairs cannot be accurately
distinguished from unrelated pairs in any populations without very large false negative rates
and in very modest-sized studies with very few pairwise comparisons.  

In a reasonably large population, the vast majority of pairs of individuals are expected to
be unrelated, or at least, effectively unrelated, sharing common ancestors only many generations
in the past.  However, a small fraction of pairs will be more recently related, and distinguishing these
kin pairs from target relationships (like PO and FS) must be accounted for.  `CKMRsim' provides
facilities for assessing error rates between different kin groups, and results for these are
summarised in (Fig.~\ref{fig:fprs}b). Here, again, focusing on the results for the genetically
depauperate SRW, and using an FNR of 0.1 we see that the chance of incorrectly
categorizing an HS as an FS,  or an FS as a PO is less than 1 in 100. Likewise, 
we can expect PO to be misidentified as FS at a rate of around 1 in 500, and 
HS or Aunts/Uncles to be misidentified as parents at a rate less than 1 in 1000.
Though these rates are much higher than the FPRs for unrelated pairs, they should be
compared to the number of actual non-target kin pairs expected.  For example, in a salmon
population, the number of full aunts or uncles of a fish is unlikely to exceed 50, 
at later life stages,  so a FPR of  $<$0.001 is comfortably low in that case.  

The presence of multiple alleles at many of these loci improves power for
kin inference compared to a marker panel using only a single SNP at each
amplicon.  To provide a visual display of the difference afforded by using
microhaplotypes compared to single SNPs in this data set, we show the
distribution of log-likelihood ratios for different relationships using our microhaplotype-scored
amplicons, versus using the most heterozygous single SNP from each
amplicon (\ref{fig:ckmr-comp}).  The figures show a considerable increase
in separation between the relationships categories from calling genotypes
as microhaplotypes in these amplicons.

While there is not a currently available method (like that of \citeauthor{anderson2006power}) for estimating error rates in
inference of parent-offspring trios with multiallelic marker data, given the substantial amount of statistical power for identifying PO and FS pairs, which 
require much more statistical power, it is clear that PO-trio inference with these markers will be essentially error free.








\section*{Discussion}

We describe a novel panel of microhaplotype genetic markers for Chinook salmon
in California that provides sufficient power for highly accurate identification 
of parent and offspring pairs and full siblings, and also allows near-perfect
identification of individuals to population or genetic group of origin in California. 
This set of genetic markers includes multiallelic gene regions with high variability for
relationship inference, gene regions identified in whole-genome sequence data
for increased power for identification of specific populations and incorporates markers that have
long been in use for both GSI and PBT in California (Clemento et al. 2014), by converting them
into so-called 'SNPlicons'. This set of markers lays the foundation for a comprehensive 
genetic monitoring and evaluation effort that facilitates multiple types of inference 
and is flexible and extensible.


This marker set and baseline reference dataset provide
unprecedented power for identifying fish from all of the Chinook salmon ESUs in California,
as well as individual populations within those ESUs (Fig.~\ref{fig:gsi}).
The California Central Valley (CCV) is one of the largest river
basins on the west coast of North America and drains the Sierra Nevada and southern Cascade
mountain ranges. It has the highest diversity of recognized Chinook salmon
ecotypes in the species range, with four named ecotypes, two of which are protected
under the US Endangered Species Act. It is also the source of water for several major urban areas as well as the
largest and most valuable agriculture area in North America. As such, accurate identification of the distinct ecotypes of CCV Chinook salmon is of utmost importance for monitoring and
evaluation of individual ecotypes, as well as designing and implementing
effective conservation and management actions. This has been challenging,
given the recent common ancestors of these ecotypes \citep{clemento2014evaluation}, and
ongoing migration between subbasins where different ecotypes predominate.

We describe the first set of genetic markers that produces easily replicable data and
that identifies all of the ecotypes in the CCV, including the late-fall-run Chinook salmon ecotype.
The late-fall-run occurs only in the CCV and shares a genomic background with the more
common CCV fall-run salmon ecotype, so has been refractory to previous GSI efforts with other marker types \citep{seeb2007development,clemento2014evaluation,thompson2024genomics}. Moreover, we show how the FRH spring-run lineage is
easily identifiable through a combination of traditional GSI and the characterization
of functional genetic markers in the RoSA. Finally, although previous work has described
GSI capabilities that distinguish the natural-spawning CCV spring-run lineages from
each other and their fall-run counterparts with moderately high accuracy, we demonstrate
near complete accuracy in distinguishing these 'stocks'. Moreover, the few fish that are
apparently misidentified (Fig. 3) are likely to be primarily migrants and not true misidentifications.
For example, the three fish that were field-characterized as spring-run from Butte Creek, but are
genetically identified as winter-run salmon clearly carry the winter-run genomic background and
are strays from the winter-run stock, as these could not realistically be misidentified on the 
basis of the genetic data.

Results from the model-based clustering analysis with the program STRUCTURE, revealed patterns 
of population relationships that are coincident with previous work (Clemento et al. 2014; 
Kinziger et al. 2013), emphasizing the distinction between Chinook salmon populations 
in Coastal California and the Central Valley. These analyses also uncover some additional patterns, 
including the clear presence of mixed ancestry in the Coastal California population in the 
Russian River, which is consistent with its location, at the southernmost edge of the 
Coastal Chinook salmon distribution and proximate to the mouth of the Sacramento River 
(the Golden Gate). In the Central Valley, the clustering results emphasize the genetic 
distinction of the SWR population, which is likely at least partially due to 
the extreme bottleneck that it passed through 15-20 generations prior. The inclusion 
of the LFAR markers also resulted in the clear distinction of the CHLF group, which 
is coincident with the long-known phenotypic distinction of this population, but represents a novel 
genetic result. Moreover, at K = 7, the fall-run reporting group breaks into two clusters, 
which are distributed across almost all of the spring and fall-run populations in the 
Central Valley, albeit not equivalently, emphasizing both current and historical migration and 
gene flow between them. 

As sequencing costs drop, whole genome sequence (WGS) data may become the preferred
data type for salmon genetics. If so, it may become possible to include
all >200 genetically distinct Chinook salmon populations within a single standardized reference baseline
constructed with WGS data
that performs equally well at broad and regional scales \citep{desaixINPRESSpopulation}.
Such an approach would be highly flexible and extensible, as it would allow for the use
just about any marker type, including reduced representation DNA sequencing (e.g. RADseq,
\citealt{meek2020identifying,thompson2024genomics}),
as regional monitoring datasets would not need to be replicable.  For the present, however,
baselines tailored to specific regions are essential for regional management questions and
targeted sequencing approaches have proven to be the most practical for large-scale
applications of either GSI, PBT or an integrated monitoring program \citep{beacham2021parentage}.

%\comment{Maybe discuss: A large fraction of the Feather River
%Hatchery Spring-run (FRHS) collection into a cluster (colored orange in Fig.~\ref{fig:struct}).
%A large fraction of the samples from the fall-run hatchery
%program at the Feather River (FRHF) are also members of that same orange cluster, while others
%belong to a cluster composed of other fall-run Chinook salmon. These STRUCTURE results at $K=7$,
%validate the finding that all populations outside of the }

%Although previous work using genome-scale data has shown that many of
%these populations or stocks can be resolved \cite{thompson2020complex,meek2020identifying},
%such data are not practical for most large-scale applications of GSI and PBT, particularly because
%of the much higher costs, longer turnaround time and challenges of producing highly
%replicable datasets with reduced representation and low-coverage whole genome sequencing methods \cite{ali2016rad}.


Previous genetic marker sets for California Chinook salmon had sufficient power for inferring
parent-offspring relationships, but only when both parents were sampled and genotyped.
This marker set considerably increases the capacity for relationship inference in California
salmon, by providing sufficient power for parent-offspring relationship inference, when only
one parent is sampled, as well as identifying pairs of full siblings when no parents are sampled.
Groups of full siblings larger than two are identified even more accurately than expressed in Fig.~\ref{fig:fprs}.
Although some additional errors might be
expected when related but non-target kin pairs are sampled,  the false positive rates
for these non-targe kin pairs are sufficiently small
for almost all realistic scenarios.
Accurate relationship
inference in even the most genetically depauperate Chinook salmon population in California
and rangewide \citep{seeb2007development,clemento2014evaluation} is therefore possible with this marker set.
We note that, as in previous work, it is nearly impossible to identify HS pairs accurately
with data from this, or any standard marker dataset. Accurate identification of HS pairs
typically relies on many hundreds of microhaplotype markers \citep{baetscher2018microhaplotypes} or thousands of SNP markers \citep{hillary2018genetic}. 

