%!TEX root = main.tex

\section*{Introduction}

Genetic markers have long been used in the 
management of Pacific salmon, and salmon conservation and management have been
behind a number of advances in molecular ecology,
from data generation techniques \citep{clemento2011discovery,campbell2015genotyping,mckinney2017managing}
to statistical methodology
\citep{smouse1990genetic,anderson2002model,pella2006gibbs}.
Two broadly applicable innovations that have been
actively fostered by the Pacific salmon research community are genetic
stock identification
(GSI:~\citealt{milner1982genetic,beacham2004dna,seeb2007development})
and parentage based tagging
(PBT:~\citealt{anderson2006power, garza2007large, abadia2013large, steele2013validation}).  



 In the 1980's, electrophoretically
detectable genetic variation, in the form of allozymes
\citep{ayala1972allozymes,allendorf1981use}, was used to
establish a program of GSI for Chinook salmon,
{\em Oncorhynchus tshawytscha} \citep{milner1982genetic}.  Extensive sampling
revealed that these allozyme markers
exhibited different allele frequencies among major stocks of Chinook salmon on the West Coast.
These allele frequency differences make GSI possible, and
\citet{milner1985genetic} soon showed that proportions of stocks in
the Washington state coastal troll fishery could be estimated by GSI.
Since that time, with the development of novel molecular markers, and, now
with increasing capacity to sequence genomic material, the scope and scale of GSI
has expanded considerably.

By using greater numbers of more variable markers than the allozymes available
in the 1980s it is possible to accurately
identify the population of origin of individual fish, rather than simply estimating
aggregated stock proportions.  It is also possible to resolve populations of fish that
are much more closely related than before.  Furthermore, reference data sets with genotypes
from hundreds of populations throughout the range of multiple species of salmon and
other anadromous species
\citep{seeb2007development,gilbey2018microsatellite,barclay2019genetic} now exist, and are routinely used to assign fish caught thousands of
kilometers from their natal streams to their stock of origin. Applications include estimating fishery
composition \citep{satterthwaite2015stock}, providing real-time information for genetics-informed fishery closures \citep{beacham2004dna}, assessing the spatial distribution of different stocks in the
ocean \citep{urawa2009stock} and their temporal distribution in upstream migrations
\citep{hess2014monitoring},  and monitoring  bycatch \citep{hasselman2016genetic} or illegal captures \citep{wilmot1999origins} in marine fisheries.

Although sequencing costs continue to decline, they are high enough
that there remains a tradeoff
between reference baselines that include information from a large number of
populations across a broad scale, and those that have been tailored
to distinguish between closely related populations on a smaller, regional scale.
Because of cost considerations, reference baselines that include populations across a broad 
spatial scale may include only a few populations from each sub-region.  Furthermore,
baselines tailored to a specific region often assemble markers that specifically
show allele frequency differences between the closely related---and hence difficult
to resolve---stocks within the region.  Consequently, regionally targeted baselines typically
outperform broad-scale baselines in resolving populations within the region.

The ascendance over the last decade of parentage based tagging (PBT) as a management
tool for Pacific salmon adds another factor to consider when developing a GSI baseline.
Since the first proposal \citep{anderson2005description} to replace or augment the coded-wire tag
programme \citep{nandor2010overview}
with PBT, it has been noted that one of the major advantages of a genetic program for
PBT is that the genetic markers used for PBT could also be useful for GSI (and
vice versa).

Here, we present a reference baseline for Chinook salmon, targeted to the complex population 
structure within the Central Valley of California. Chinook salmon of the two main river basins---the 
Sacramento and the San Joaquin---within the Central Valley exhibit the greatest run-timing diversity
within the species.  With four recognized ecotypes, delineated primarily on the basis of run timing
(fall-, late-fall-, winter-, and spring-run), adult Chinook salmon can be found migrating or residing
in freshwater every month of the year in California. \comment{NEED MORE HERE. Certainly we should cite
\citet{yoshiyama1998historical}.}


 \section*{Methods}

\comment{More verbiage}

\subsection*{Population Sampling}

We sampled fish for the reference baselines from XX locations across California.
These locations included sites in the two
main tributaries (the Sacramento River and the San Joaquin River) within
California's Central Valley, the Klamath-Trinity basin in Northern California, and
3 rivers of the California coast (Russian, Eel, etc)
(Fig.~\ref{fig:map}).
%%%%%%%%%%%%%
\begin{figure}
\newcommand{\mapcap}{\footnotesize  A map of sampling locations represented in the reference 
baseline.  The colored circles aside
each location name represent the different run-timing groups of Chinook salmon
included in the collections from the location.  Color codes are as follows: \Wball~=~Winter run,
\LFball{}~=~Late-fall run, \Fball~=~Fall run, \Sball~=~Spring run. 
Names of each location are followed by the codes used to refer to fish from each location, with
{\tt XXF/S} indicating {\tt XXF} and {\tt XXS} are used to denote fall-run and spring-run fish
from location {\tt XX}.  Abbreviations used in location names as follows: R. = River; Ck. = Creek; H. = Hatchery; NFH = National Fish Hatchery.}
\begin{center}
\includegraphics[width=\columnwidth]{images/map-crop.pdf}
\end{center}
\caption[\mapcap]{\mapcap}
\label{fig:map}
\end{figure}
%%%%%%%%%%%%%%%
Collections in each location were separated into fish from the different run-timing ecotypes according
to a variety of criteria. The last vestiges of winter-run Chinook salmon are maintained at the Livingston
Stone National Fish Hatchery for supplementation (released as juveniles into the river) and as a captive broodstock;
sampling of this population was performed by hatchery staff during spawning. In Central Valley Rivers with both 
spring and fall ecotypes (i.e. Mill, Deer and Butte Creeks), previous research has documented distinct/disparate/unique
spawning times (Julian date) for each ecotype (REF?). As such, sampling targeted carcasses found in each of these 
rivers at times representative of their run-timing designation, which we were able to further verify using their genotypes
at the RoSA markers. At the time of sampling, only fall-run Chinook were found in the San Joaquin River. Run-timing 
designation at the Feather River Hatchery is more complicated but is managed at a fine-scale by hatchery practices, 
where early returning fish in May and April (expressing the Spring-run life history) are visibly tagged and re-released 
to the river. Subsequently in the Fall, when hatchery raceways are reopened, only fish with the visible tag are included
as Spring-run broodstock. Individual-based analyses of RoSA genotypes have demonstrated 

(We need to talk about how these were split into spring run, fall-run, winter 
run, etc.
In total, the samples in the baseline represent fish from five different Evolutionarily Significant 
Units (ESUs) of Chinook salmon \citep{waples1991pacific} A summary of the different collections 
%%%%%%%%%%%%%%%%
\begin{table*}
\caption{\footnotesize Collections of fish in the reference baseline.  The Sampling Months column shows
a pictorial representation of the proportion of each collection sampled in each month.  Colors in
this column correspond to run-timing groups as described in the caption of Fig.~\protect\ref{fig:map}.}
\label{tab:samples}
{\footnotesize 
\begin{tabular*}{\linewidth}{@{\extracolsep{\fill}} llllllrcr}
\hline\hline\vspace*{0.4ex}
\input{inputs/samples-table.tex}
\end{tabular*}
}
\rule{3cm}{0.3pt}

{\footnotesize
$^1$R. = River.\\
$^2$Ck. = Creek; NFH = National Fish Hatchery; R. = River; H. = Hatchery.\\
$^3$ESU = Evolutionarily Significant Unit; CVFR = Central Valley Fall Run ESU; CVSR = Central Valley Spring Run ESU; CVWR = 
Central Valley Winter Run ESU; CC = California Coastal ESU; UKTR = Upper Klamath and Trinity Rivers ESU. \\
$^4N_\mathrm{day}$: the number of samples for which the specific sampling day was recorded.
}
\hrule\vspace*{0.3ex}\hrule
\end{table*}
%%%%%%%%%%%%%%%%%%%%%

\subsection*{Genetic Markers}

To assemble the markers for this reference baseline, we drew upon several different
sources, as detailed in the separate sections below.

\subsubsection*{Conversion of existing SNPtype assays}

A number of SNP markers from earlier discovery efforts \citep{clemento2011discovery} had already 
been validated as useful for GSI
and PBT in California \citep{clemento2014evaluation}. These markers were originally assayed
using TaqMan (Life Technologies Corp., Carlsbad, CA) probes or SNPtype assays \comment{(CITE)}.
In order to type these markers using amplicon sequencing, we followed the methods in
\citet{campbell2015genotyping} and added the Illumina small RNA sequencing primer and 
read 2 sequencing primer to the SNPtype Specific Target Amplification Primer and Locus 
Specific Primers respectively.  We then dropped any loci that produced genotypes that were 
frequently inconsistent with genotypes produced with the SNPtype assays and any loci that 
did not amplify well or those that amplified so well that they were grossly over-represented
among the reads on the sequencer.  We then 
used Freebayes \citep{garrison2012haplotype} to identify all variants on the fragment (not just the SNPs targeted by the original TaqMan assays).  This allows us to score each locus as a
microhaplotype \citep{baetscher2018microhaplotypes} when multiple variants occur in the sequence.  We filtered out variants
that were considered multinucleotide or complex variants by Freebayes.

\subsubsection*{Additional microhaplotype discovery}

We undertook a new discovery process to develop additional multiallelic
microhaplotype \citep{baetscher2018microhaplotypes} markers in California Chinook.
\comment{NEED TO DESCRIBE. How did we look for these, etc. What constituted a
candidate to then follow up on with amplicon sequencing, etc.}

After designing and validating an amplicon-sequencing approach to type both the
existing SNPtype assays and the product of the additional microhaplotype
discovery process we identified their location with the genome. We mapped the reference
sequences for these markers 
to the Chinook v2.0 genome assembly (\url{https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_018296145.1/})
with both the Burrows-Wheeler Aligner \citep{bwa-mem2009} and the BLAST-like Alignment tool \citep{kent2002blat}.
We report the genomic coordinates these target amplicons found with both alignment methods.

\subsubsection*{Identifying and Designing markers from the Late-Fall Associated Region}

To identify loci with large allele frequency differences between late-fall and fall-run Chinook salmon, 
we first conducted a simple case-control GWAS using the low-coverage whole genome sequencing 
data of \citet{thompson2020complex}.  We conducted two separate association-study comparisons.  
The first involved the 16 late-fall run fish as cases with 16 fall-run fish from the Feather River Hatchery as controls. 
The second involved the same 16 late-fall fish as cases, but used 16 fall-run fish from the San 
Joaquin River as controls.  The association studies were performed for each chromosome with 
ANGSD \citep{pmid21663684,korneliussen_angsd_2014}, using the following command: {\footnotesize\tt angsd -yBin \{ybin\}  -r \{chrom\} 
-minMapQ 30 -minQ 20 -minInd 12 -doAsso 1 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP\_pval 1e-6  
-out \{prefix\}  -bam \{bamlist\}}, where {\tt \{ybin\} } is the path to a file indicating the cases and 
controls, 
{\tt \{chrom\} } indicates the chromosome name, {\tt \{prefix\} } is the prefix to use for output files, and 
{\tt \{bamlist\} } is the path to the text file holding paths to the BAM files, one for each individual, ordered as in {\tt \{ybin\} }.

There was only one peak in association scores that was shared between both comparisons (see 
Results). From this peak we identified candidates for followup genotyping by including the 10 
SNPs with the lowest association p-values, and supplementing those with additional SNPs that 
showed:
\begin{itemize}
\item An estimated absolute allele frequency difference (between late-fall and all 32 fall-run fish)  $>0.75$,  with an approximate lower confidence interval of that figure of 0.5,  or
\item An estimated absolute frequency difference of >0.5 with a lower confidence interval >0.25 and 
with either fall-run or late-fall run being nearly fixed for one of the alleles, or
\item An annotation from snpEff \citep{cingolani2012program} of “HIGH” or “MODERATE”, or
\item An absolute frequency difference $>0.48$ and a lower confidence interval of $>0.25$ and a 
genomic  coordinate $> 2.2$~Mb.
\end{itemize}
The final condition was implemented to gather several SNPs within the flanking region to the right of 
the main peak of association. 

For the candidate markers found by the above process, we designed amplification primers using 
Primer3 \citep{koressaar2007enhancements,untergasser2012primer3} and tested amplification in a 
subset of Chinook salmon 
samples. The primer pairs that amplified sequence that reliably mapped back to the association-peak region were then typed
in a separate collection of 1,638 Sacramento River Chinook salmon sampled from the fish trap below Keswick Dam, including 1,169 winter run, 281 spring 
run, and 188 fall/late-fall run. We divided the fish identified as fall/late-fall via the 125-locus genetic 
stock identification data set into fall run and late fall run according to their sampling date at Keswick: 
99 fish arriving before Dec 26 were classified as fall run and 89 fish arriving after Dec 26 were 
classified as late-fall run.   The allele frequencies of the target SNPs at the remaining amplicons 
were 
estimated for each of these four groups: winter, spring, fall, and late-fall, and we retained
amplicons showing large allele frequency  differences between late-fall fish and all other runs.


\subsubsection*{Markers around the RoSA region of \citet{thompson2020complex}}

We include amplicons targeting SNPs within the ``Region of Strongest Assocation'' (RoSA)
between spring-run and fall-run fish described in \citet{thompson2020complex}.
\comment{Need more here. I personally am of the thought that we should not include these
for GSI, but should use them for identifying the genotypes of fish assigned to populations
using the rest of the markers.}

\subsection*{Population genetic analyses}

\subsection*{Power for genetic stock identification and population assignment}

\subsection*{Power for relationship inference}



\section*{Results}

\subsection*{Genetic Markers}

\subsubsection*{Conversion of existing SNPtype assays}

Of \comment{XX} existing SNPtype assays, we were able to convert \comment{YY}
into reliable amplicon-sequenced assays that were included in the panel.

\subsubsection*{Additional microhaplotype discovery} 

We tested \comment{XX} candidates from our additional microhaplotype discovery and
retained \comment{185} of those in the panel.

Of these \comment{185} loci that originated from SNPtype assays or from our additional
microhaplotype discovery, \comment{179} of them mapped identically to a single
location on an assembled chromosome in the Otsh\_v2.0 genome, using both bwa mem and BLAT.
Of the remaining 6 loci, one of them was mapped to the same location by BLAT and bwa mem, but
the alignments differed by one base pair in the length of an embedded insertion, two of them had a single secondary alignment on a different chromosome
that was identical in both bwa mem and BLAT, two of them had secondary alignments that
disagreed slightly between BLAT and bwa mem, and one of them had multiple
non-primary alignments (\comment{SuppTable}).

\subsubsection*{Late-fall Associated Region Markers}

The case-control association test revealed a single peak on chromosome 34 with large allele frequency differences
between Central Valley late-fall and fall-run fish.  This peak was evident both in the comparison of
late-fall fish to fall-run fish in the Feather River Hatchery and of late-fall fish to the fall-run fish in
the San Joaquin River (Fig.~\ref{fig:lfar-assoc}).
%%%%%%%%%%%%%%%%%%%%%
\begin{figure*}
\newcommand{\lfarcap}{\footnotesize Negative log base 10 of association $p$-values for
individual SNPs for late-fall versus fall run.  $x$ axis shows position in genome (in megabases),
with color alternating by chromosome, as indicated by numbers above the $x$-axis. ``Unk'' refers
to unplaced scaffolds in the Otsh\_v1.0 genome assembly \citep{christensen2018chinook}. The 
upper panel is the comparison between late-fall and Feather River Hatchery fall, while the lower 
panel is the comparison of late-fall to San Joaquin River fall. }
\includegraphics[width=\textwidth]{images/lfar-assoc-faceted.jpg}
\caption[\lfarcap]{\lfarcap}
\label{fig:lfar-assoc}
\end{figure*}
%%%%%%%%%%%%%%%%%%%%
The Manhattan plot of Fig.~\ref{fig:lfar-assoc} shows sporadic SNPs with low association
$p$-values.  These are not shared between the two different late-fall to fall comparisons, and
sites that we investigated were concentrated in areas of poor mapping suggesting that their
$p$-values were artifactual.  By contrast, a large number of sites near the prominent peak on
chromosome 34 had large allele frequency differences between late-fall and fall run.  Taking the
top 10 most associated SNPs, and including additional ones from the filtering criteria given in the
Methods, yielded 49 candidate markers for which we designed primers for amplication.  Only 8
of these 49 primer pairs yielded reliable amplification and mapping.   Of these 8, two of the SNPs 
showed inconsistent genotyping and were removed from consideration.  Of the remaining 6, only 3 
showed marked allele frequency differences ($> 0.7$) between late-fall and fall run.  These three 
SNPs (Chr34:828,768,  Chr34:865,057, and Chr34:1,063,084) also had pronounced differences in 
allele frequency between 
late-fall and spring or winter run.

Analysis of the mapping of amplicons for Chr34:828,768 
showed that many ($\approx$89\%) of the reads were off-target (aligning to other chromosomes, 
etc.), making it costly (in terms of sequencing effort) to include the marker in the baseline panel. 
Consequently it was dropped.  The frequency of the different alleles at the two remaining loci
within different reporting units in the baseline are shown in Table~\ref{tab:lfar-freqs}.
%%%%%%%%%%%%%%
\begin{table}
\caption{\footnotesize Allele frequencies across reporting units of the two late-fall associated
markers.  Frequencies are given for the alleles most common amongst late-fall Chinook: nucleotide 
{\tt G} at locus Chr34:865,057 and {\tt A} at Chr34:1,063,084.  $N$ is the total number of gene 
copies available for estimating the allele frequency.}
\label{tab:lfar-freqs}
{\footnotesize
\begin{tabular*}{\columnwidth}{@{\extracolsep{\fill}} lrrrr}
\hline\hline
& \multicolumn{2}{c}{\underline{Chr34:865,057}} & \multicolumn{2}{c}{\underline{Chr34:1,063,084}} \\
Reporting Unit & freq. & $N$ & freq. & $N$ \\ \hline
\input{inputs/lfar-freqs.tex}
\end{tabular*}
}
\vspace*{-2.3ex}\hrule\vspace*{0.3ex}\hrule
\end{table}
%%%%%%%%%%%%%%%
\subsection*{Population genetic summaries}

Barplot of the total number of alleles per locus in the whole data set
is Figure~\ref{fig:num-alle}.
%%%%%%%%%%%%%%%%%%%%%% 
\begin{figure}
\newcommand{\numacap}{\footnotesize Number of loci with different
total numbers of alleles in the data set.}
\begin{center}
\includegraphics[width=\columnwidth]{images/num-alle-barplot.pdf}
\end{center}
\caption[\numacap]{\numacap}
\label{fig:num-alle}
\end{figure}
%%%%%%%%%%%%%%%%%%%%

And the table of the popgen summary is Table~\ref{tab:pg-summ}.
%%%%%%%%%%%%%%
\begin{table}
\caption{\footnotesize Simple population genetic summaries. All quantities are means
over loci.  $\bar{N}$ is the number of fish with observed genotypes; $\bar{N}_\mathrm{A,ss}$ is the
average number of alleles and $\bar{P}_\mathrm{poly,ss}$ the fraction of polymorphic loci,
after sub-sampling to the smallest sample size per locus; $\bar{H}_\mathrm{exp}$ and
$\bar{H}_\mathrm{obs}$ are expected and observed heterozygosity, respectively. Population
codes are as given in Table~\ref{tab:samples}}
\label{tab:pg-summ}
{\footnotesize
\begin{tabular*}{\columnwidth}{@{\extracolsep{\fill}} lrrrrr}
\hline\hline
\input{inputs/popgen-summary.tex}
\end{tabular*}
}
\vspace*{-2.3ex}\hrule\vspace*{0.3ex}\hrule
\end{table}
%%%%%%%%%%%%%%%



\subsection*{Power for genetic stock identification and population assignment}

Results of power analysis for GSI and also Fst estimates are in Figure~\ref{fig:gsi}
%%%%%%%%%%%%%%%%%%%%%% 
\begin{figure}
\newcommand{\gsicap}{\footnotesize Self-assignment rates  ({\bf a)} and pairwise 
$F_\mathrm{ST}$ values and allele frequencies ({\bf b)} for the collections in the baseline.  The codes of the populations
appear on the outer margins of the table atop cells colored according to the run-timing group
of the collection.  Interior cells in which the column and row correspond to the same reporting
unit are colored according to the reporting unit.  For self-assignment tallies, the row corresponds
to the true source collection and the column corresponds to the assigned collection. In ({\bf b}), for $F_\mathrm{ST}$ values, the upper triangle was computed using the baseline microhaplotype
data. Each cell in the lower triangle shows a small scatter plot of estimated allele frequencies
in the collections---the $x$ value is allele frequencies from the collection in the column and the
$y$ value is from the collection in the row. Frequencies for all alleles are shown, with values within
each cell ranging from 0 to 1.  These frequencies were estimated by, for each locus, randomly
subsampling individuals from each collection to the minimum number of observed alleles in any collection
at the locus.  }
\begin{center}
\includegraphics[width=\columnwidth]{images/gsi_and_fst_fig-crop.pdf}
\end{center}
\caption[\gsicap]{\gsicap}
\label{fig:gsi}
\end{figure}
%%%%%%%%%%%%%%%%%%%%







\subsection*{Power for relationship inference}

The results are summarized in Fig.~\ref{fig:fprs}.
%%%%%%%%%%%%%%%%%%%%%
\begin{figure*}
\newcommand{\fprcap}{\footnotesize False positive rates as a function of false negative
rates for distinguishing between different relationships.  {\bf (a)} False positive rates calculated
using importance sampling of unrelated pairs of individuals being incorrectly categorized as parent-offspring (PO),
full-sibling (FS) or half-sibling (HS) pairs.  {\bf (b)}  False positive rates calculated using regular Monte Carlo
sampling of avuncular pairs (AN: aunt-niece, uncle-nephew, etc.) or full-siblings (FS) or half-siblings (HS) being 
incorrectly categorized as parent-offspring (PO); as well as FPRs of parent-offspring (PO) pairs or half-sibling (HS) 
pairs being incorrectly inferred as full siblings (FS).  In both {\em a} and {\em b}, 100,000 Monte Carlo replicates 
were simulated.  Importance sampling allows estimation of very small false positive rates for unrelated pairs, but 
cannot be applied to truly related pairs.  Consequently, it is difficult to accurately estimate small probabilities 
($10^{-4}$ or less) in {\bf b}, as indicated by the error bars, which represent the estimate $\pm 2s$ where $s$ is the estimated standard error of the mean of the Monte Carlo sample.  }
\begin{center}
\includegraphics[width=\textwidth]{images/fpr-fnr-figure-crop.pdf}
\end{center}
\caption[\fprcap]{\fprcap}
\label{fig:fprs}
\end{figure*}
%%%%%%%%%%%%%%%%%%%%





\section*{Discussion}

When, and if, sequencing costs drop sufficiently, it may be possible to include
all different populations of a single species within a single standardized reference baseline
that performs equally well at broad and regional scales. A statistical approach for population
assignment from low coverage whole genome sequences appeared recently
\citep{desaixINPRESSpopulation}, and its authors noted that such a data type could
be ideal for producing reference baselines simultaneously applicable to broad scale
coverage and high resolution within sub-regions.   For the present, however,
baselines tailored to specific regions are essential for regional management questions.
It should be noted that recent 

