---
title: "LFAR and WRAP Allele Freqs"
author: "Eric C. Anderson"
date: "Last Updated: `r Sys.Date()`"
output:
  html_document:
    df_print: paged
    toc: true
    toc_float: true
bibliography: "`r here::here('tex/citation.bib')`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE)
knitr::opts_knit$set(root.dir = here::here())
start_time <- Sys.time()
```

# Input and Output Paths

```{r}
if(exists("snakemake")) {
  input_list <- snakemake@input
  output_list <- snakemake@output
} else {
  # inputs:
  input_list <- list(
    final_baseline = "data/SWFSC-chinook-reference-baseline.csv.gz",
    pop_labels = "inputs/reference-collection-names.csv",
    locus_info = "inputs/Calif-Chinook-Amplicon-Panel-Information.csv"
  )
  # outputs:
  output_list <- list(
    lfar_tex_table = "tex/inputs/lfar-freqs.tex",
    wrap_tex_table = "tex/inputs/wrap-freqs.tex"
  )
}
# you can create the necessary output directories like this:
dump <- lapply(output_list, function(x)
  dir.create(dirname(x), recursive = TRUE, showWarnings = FALSE)
)
```



# Introduction

The goal here is simply to get the allele frequenccies in different reporting units
at the LFAR markers into a table format.  


First, get the baseline and attach the collection and repunit names that we want.
```{r, message=FALSE, warning=FALSE}
library(tidyverse)



# get the full baseline with the new markers, etc.  We will toss out some
# non-winter Sacto fish of unknown provenance
full_base <- read_csv(input_list$final_baseline)


# read the pop labels for proper sorting
pop_labels <- read_csv(input_list$pop_labels)

# rename those collections and repunits
full_base2 <- full_base %>%
  rename(old_name = collection) %>%
  select(-repunit) %>%
  left_join(pop_labels %>% select(-run_timing_group), by = join_by(old_name)) %>%
  select(-old_name) %>%
  select(indiv, repunit, collection, sample_type, everything())
  
# finally, get the order of collections and repunits we want
repunit_order <- rev(unique(pop_labels$repunit))
```

Now, whittle that down to just the LFAR and pivot it longer:
```{r}
lfar_base <- full_base2 %>%
  select(indiv:collection, starts_with("Ots_lfar")) %>%
  pivot_longer(
    cols = starts_with("Ots_lfar"), 
    names_to = c("locus", "gene_copy"),
    names_pattern = "^(.*)_([12])",
    values_to = "allele"
  )
```

And now we count up the alleles within repunits
```{r}
# first, get the multiallelic counts
lfar_cnts_MA <- lfar_base %>%
  filter(!is.na(allele)) %>%
  count(repunit, locus, allele)

# now we want to pick out the actual bases that are relevant and count the up
lfar_cnts <- lfar_cnts_MA %>%
  mutate(
    base = case_when(
      locus == "Ots_lfar001_34_00954022" ~ str_sub(allele, 1, 1),
      locus == "Ots_lfar002_34_01151770" ~ str_sub(allele, 2, 2)
    )
  ) %>%
  group_by(repunit, locus, base) %>%
  summarise(n = sum(n)) %>%
  ungroup() %>%
  complete(repunit, locus, base, fill = list(n = 0L)) %>%
  group_by(repunit, locus) %>%
  mutate(
    tot_n = sum(n),
    fract = n / tot_n
  ) %>%
  ungroup()

lfar_cnts
```

Now, we just want to focus on the alleles that are most frequent in the late-fall.
```{r}
focal <- lfar_cnts %>%
  filter(repunit == "CV-Late-Fall") %>%
  group_by(locus) %>%
  filter(n == max(n))

alle_tib <- lfar_cnts %>%
  semi_join(focal, by = join_by(locus, base)) %>%
  mutate(
    repunit_f = factor(repunit, levels = repunit_order),
    f = sprintf("%.3f", fract)
  ) %>%
  select(-repunit, -base, -n, -fract) %>%
  pivot_wider(names_from = locus, values_from = c(f, tot_n)) %>%
  arrange(repunit_f) %>%
  select(repunit_f, `f_Ots_lfar001_34_00954022`, `tot_n_Ots_lfar001_34_00954022`, `f_Ots_lfar002_34_01151770`, `tot_n_Ots_lfar002_34_01151770`)

alle_tib
```


Now, we will print this out without any column names so that we can use it in a TeX table.
```{r}
write_delim(alle_tib, delim = "&", eol = "\\tabularnewline\n", file = output_list$lfar_tex_table, col_names = FALSE)
```


And, as a final hurrah, we need to get the actual positions of those bases in Otsh_v2.0
in order to make the TeX column headers.

```{r}
locus_info <- read_csv(input_list$locus_info)

lfar_info <- locus_info %>%
  filter(str_detect(AmpliconName, "lfar"))

# for Ots_lfar001_34_00954022 the focal SNP is the first one
# for Ots_lfar002_34_01151770 the focal SNPs is the second one

# the positions in the amplicons are like this:
lfar_info %>%
  select(AmpliconName, SNPLocationsInAmpliconSequence)
```


So, 33 for the first and 66 for the second. And we can get the SNP positions
in the genome like this:
```{r}
lfar_info %>%
  mutate(
    Otsh_v2.0_snp_pos = case_when(
      AmpliconName == "Ots_lfar001_34_00954022" ~ AmpliconStartPos + 33 - 1,
      AmpliconName == "Ots_lfar002_34_01151770" ~ AmpliconStartPos + 99 - 1
    ),
    .after = Otsh_v2.0_chromosome
  )
```

So, we put those in the header of the TeX table for these.


# WRAP allele frequencies

Here we do something similar to what we did above, but we use the WRAP.
```{r}
wrap_base <- full_base2 %>%
  select(indiv:collection, starts_with("Ots_wrap")) %>%
  pivot_longer(
    cols = starts_with("Ots_wrap"), 
    names_to = c("locus", "gene_copy"),
    names_pattern = "^(.*)_([12])",
    values_to = "allele"
  )
```

Then count up the alleles in each reporting group:
```{r}
wrap_freqs <- wrap_base %>%
  filter(!is.na(allele)) %>%
  count(locus, repunit, allele) %>%
  group_by(locus, repunit) %>%
  mutate(freq = n / sum(n)) %>%
  ungroup()
```

Now, for properly sorting the alleles, let's get the freqs in Winter-Run for each.
```{r}
alle_ord <- wrap_freqs %>%
  filter(repunit == "CV-Winter") %>%
  arrange(locus, desc(freq)) %>%
  group_by(locus) %>%
  mutate(order = 1:n()) %>%
  ungroup() %>%
  select(locus, allele, order)
```

This turns out to be all the alleles, too.  Good.

Now we can format a table:
```{r}
wrap_for_table <- wrap_freqs %>%
  left_join(alle_ord, by = join_by("locus", "allele"), relationship = "many-to-one") %>%
  select(-n) %>%
  pivot_wider(names_from = repunit, values_from = freq, values_fill = 0.0) %>%
  arrange(locus, order) %>%
  select(locus:order, `CV-Winter`, `CV-Spring`, everything())

wrap_for_table
```

Write it to a table:
```{r}
wft2 <- wrap_for_table %>%
  select(-order) %>%
  extract(
    locus, 
    into = c("chrom", "pos"), 
    regex = "Ots_wrap[0-9]+_([0-9]+)_([0-9]+)", 
    convert = FALSE, 
    remove = FALSE
  ) %>%
  mutate(locus = str_c("Chr", chrom, ":", pos)) %>%
  select(-chrom, -pos) %>%
  mutate(across(.cols = -c(locus, allele), .fns = function(x) sprintf("%.3f", x)))


write_delim(wft2, delim = "&", eol = "\\tabularnewline\n", file = output_list$wrap_tex_table, col_names = FALSE)
```

  

# Citations

::: {#refs}
:::

# Session Info

```{r}
sessioninfo::session_info()
```

# Running Time

Running the code and rendering this notebook required approximately this much time:

```{r}
Sys.time() - start_time
```
