---
title: "california-chinook-microhaps: Data and code repository for \"A multipurpose microhaplotype panel for genetic analysis of California Chinook salmon\""
author: "Eric C. Anderson (also repo maintainer/owner), Anthony J. Clemento, Matthew A. Campbell, Devon E. Pearse, Anne K. Beulke, Cassie Columbus, Ellen Campbell, Neil F. Thompson, John Carlos Garza"
output: 
  github_document:
    toc: true
    toc_depth: 3
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
---

**Last Updated:** `r Sys.Date()`


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# use this for the the README.md
GPbase <- "https://eriqande.github.io/california-chinook-microhaps/"

# use this for making the README.html that we will hard link to docs/index.html
Dbase <- "./"

# choose the one that you want here depending on if you are on the
# GitHub README or are making an index.html for the docs directory
hdir <- Dbase

make_link <- function(Rmd, base = hdir) {
  nb <- stringr::str_replace(Rmd, "Rmd$", "html")
  sprintf("%s\n\n(_Compiled RMarkdown HTML document on GitHub Pages:_ [%s](%s%s))\n\n", Rmd, nb, base, nb)
}
```



This repository includes code, data, and some intermediate results to
reproduce the results in Anderson et al. ("A multipurpose microhaplotype
panel for genetic analysis of California Chinook salmon").
You can get the whole thing by cloning or downloading the repo from 
[https://github.com/eriqande/california-chinook-microhaps](https://github.com/eriqande/california-chinook-microhaps).

If you are viewing this as the README of a GitHub repository, note that
you can read it in a somewhat friendlier format on GitHub pages at:
[https://eriqande.github.io/california-chinook-microhaps/](https://eriqande.github.io/california-chinook-microhaps/)

Most of this repository concerns analyses done using R and that are fully contained
within this repo and intended to run automatically to create the figures and
tables used in the paper. 

In most cases, products from cluster computing are placed in this repository within
the `stored_results` directory.  This means that all the RMarkdown documents can be
evaluated successfully without first running the cluster-based analyses.

Analyses requiring cluster computing are documented, but do not run
automatically. 

All the RMarkdown documents can be evaluated en masse using the included
Snakefile which also shows the different output created by each notebook.
Or, if desired, they can be run one at a time in, for example,
RStudio.  Each RMarkdown document creates an HTML file in the docs/
directory that is served up publicly on GitHub pages, so that the
analyses and codes can be read in that context.

On my Mac, it takes about 10 minutes to run all the notebooks, with
002 and 003 being the ones that are most computationally intensive.

# Preliminaries and Dependencies

To the extent possible, we have tried to provide automatic installs of R package
dependencies and some public data sets.  Accordingly, you should be connected
to the internet when you run any of this.  

Running the code here requires the R programming language. It was built most
recently using R 4.4.3.  Presumably any R 4.4.x version will work.  Required R
packages (and their corresponding versions) are managed by `renv`.  If you use
RStudio then you should be prompted to install all the packages upon
opening the project.  

The R code creates publication-ready figures and makes heavy use of the `pdfcrop`
utility.  This utility should be available on your path.  It comes with most TeX
distributions, I believe.  You will need LaTeX to typeset the paper if desired.
Alternatively, you can obtain `pdfcrop` from: 
[https://ctan.org/pkg/pdfcrop?lang=en](https://ctan.org/pkg/pdfcrop?lang=en).



# RMarkdown Documents

The following RMarkdown documents should be evaluated in order.
The script `render-numbered-Rmds.R` will do that (except for `002-allele-frequencies.Rmd`)
when run in the top level
of this repository.
Some RMarkdown documents rely on
outputs from previous ones.  Some of these RMarkdown documents include
calls to Unix utilities, so might not run on non-Unix or non-Linux architectures.

Outputs (figures, tables, R data objects, etc) from each RMarkdown document
are written to the `outputs/XXX` directories.  Intermediate files
written during evaluation of each RMarkdown document are written
to the `intermediates/XXX` directories. To facilitate working between
the cluster and a desktop/laptop, some outputs are written to
the `stored_results/XXX` directories which are version controlled and
included in this repo.


Thumbnails of the figures and tables generated by each RMarkdown document appear
below.  Additionally, at the top of each section is a link to the
compiled (HTML-version) of the RMarkdown document on GitHub Pages.  


## `r make_link("001-align-coho-genome.Rmd")`

The products of this analysis go into creation of some of the trees (`009`) and also
in the designation of the alleles as ancestral or not in some of the haplotype rasters (`006`).

## `r make_link("002-allele-frequencies.Rmd")`

Calculation of allele frequency differences between early- and late-running forms
across the genome, and around GREB1L, and the generation of plots.

[![](./docs/images_and_thumbs/allele-freqs-figure_thumb.png)][alle_freqs]


## `r make_link("003-extract-johnson-creek-variants.Rmd")`

Using the previously assembled Chinook salmon genome from @narumGenomicVariationUnderlying2018
to ascertain variation from that Upper Columbia River summer/spring-run fish
around the GREB1L region.  The products of this analysis go into some later trees and
haplotype rasters.

## `r make_link("004-prepare-haplotypes.Rmd")`

Using BEAGLE to impute genotypes and infer haplotypes in 5 Mb around GREB1L.
The outputs of this analysis are used for a number of downstream analyses.

## `r make_link("005-annotating-variants-near-greb1l.Rmd")`

Using snpEff to annotate variants and then analyzing a few of them.  This
provides data that goes into Table S1 ("Two near-perfectly associated, non-synonymous variants in the GREB1L gene"),
and is also used later for the haplotype raster plots.

[![](./docs/images_and_thumbs/table-s2-2-candidate-snps_thumb.png)][snpeff_res]

## `r make_link("006-haplo-raster-plots.Rmd")`

Compiling some auxiliary information (gene/exon inclusion, etc.) and
then making haplotype raster plots using functions from the
'ecaRbioinf' package.

[![](./docs/images_and_thumbs/moderate-zoom-haplo-raster-with-JC-fish_thumb.png)][haplo_raster]
[![](./docs/images_and_thumbs/read-depth-raster_thumb.png)][read_depth_raster]

