---
title: "Renaming The Amplicons and Tidying up the Amplicon Information Table"
output: html_notebook
---


Before releasing this, we want to have a decent, consistent naming convention for
all of the amplicons.  We want a short, unique, descriptive name that will tell us
what kind of marker it is (snplicon vs microhap, etc), and then we want to have
position information in there too.  

Here is the convention that we came up with:

General pattern = `Ots_type00X_ChrNum_V2StartPos`, where:

- `Ots` is just `Ots`
- `type` is a brief descriptor (shoot for 4 characters) for example:
    + SNPtype = `scon`
    + Microhap = `mhap`
    + RoSA = `rosa`
    + SexID = `sexy`
    + LFAR = `lfar`
    + WRAP = `wrap`
    + VGLL = `vgll`
    + Six6 = `sixx`
- 00X is a three digit left zero-padded, base-1 number within type ordered by genome V2 location.
- ChrNum is a two digit left zero padded Chromosome number, except for sexy which is
  going to be just the scaffold name, with the underscore removed.
- V2StartPos is just where it maps in Otsh_V2.

### As an aside: 

To process these things in the snakemake mega-simple workflow, we will add a new
field that goes within `genome` or `target_fasta` in the config file which is called
`rename:` and that give tsv file with columns named `orig` and `new` which are the
original names and the desired new names, and we will make one extra rule in there
somehow to deal with it.  We will keep the VCFs with the original names, but we need
a rule that will make a renamed VCF, and we also need a renamed Microhaplot output.

So we won't change the target fastas.  This way we have a data set that is compatible
with old stuff, and we also have one that we can use with the new names.


## Doing it

So, let's get crackin' on slurping stuff out of the Excel file and making these modifications.

```{r}
library(tidyverse)
library(readxl)


# read it in and change some of the names
orig <- read_excel("ChinookFullPanelLociTable_May2024Updated.xlsx") %>%
  rename(
    OtherName = Locus,
    Otsh_v1.0_coordinates = AmpliconGenomeV1Location,
    Otsh_v2.0_coordinates = AmpliconGenomveV2Location,
    Alternate_Otsh_v1.0_coordinates = `AlternateGenomeV1Location(s)`,
    Alternate_Otsh_v2.0_coordinates = `AlternateGenomeV2Location(s)`,
    LocusOrigin = LocusType,
    OriginalReference = `Original Reference`,
    MappingNotes = Notes
  )

# now, reorder the columns so that more information is front loaded
orig2 <- orig %>%
  select(
    LocusOrigin,
    OriginalReference,
    OtherName,
    ForwardPrimerSequence,
    ReversePrimerSequence,
    ReferenceSequence,
    SNPLocationsInReferenceSequence,
    TrimmedAmpliconSequence,
    Otsh_v2.0_coordinates,
    Alternate_Otsh_v2.0_coordinates,
    Otsh_v1.0_coordinates,
    Alternate_Otsh_v1.0_coordinates,
    everything()
  )
```


Now we are going to get columns of chromosome number (or name if unmapped) and
start positions, so that we can order things by genome coordinate.  

```{r}
# get the chromosome numbers (pulled from NCBI page at https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_018296145.1/)
chroms <- read_tsv("Otsh_v2.0_chroms.tsv") %>%
  mutate(Otsh_v2.0_chromosome = str_replace(Chromosome, "LG", ""), .before = Chromosome) %>%
  select(RefSeq, Otsh_v2.0_chromosome)


orig3 <- orig2 %>%
  extract(Otsh_v2.0_coordinates, into = c("RefSeq", "AmpliconStartPos"), regex = "^(.*):([0-9]+)-[0-9]+$", remove = FALSE, convert = TRUE) %>%
  left_join(chroms, by = join_by(RefSeq)) %>%
  mutate(Otsh_v2.0_chromosome = ifelse(is.na(Otsh_v2.0_chromosome), str_replace(RefSeq, "_", ""), Otsh_v2.0_chromosome)) %>%
  select(Otsh_v2.0_chromosome, AmpliconStartPos, everything()) %>%
  select(-RefSeq) %>%
  arrange(Otsh_v2.0_chromosome, AmpliconStartPos)

```


Now we
will be able to rename everything all at once, easily.

```{r}
final <- orig3 %>%
  mutate(
    tmpname = case_match(
      LocusOrigin,
      "LFAR" ~ "lfar",
      "Microhap" ~ "mhap",
      "RoSA" ~ "rosa",
      "SNPtype" ~ "scon",
      "Sex ID" ~ "sexy",
      "Six6" ~ "sixx",
      "VGLL3" ~ "vgll",
      "WRAP" ~ "wrap"
    )
  ) %>% 
  group_by(tmpname) %>%
  mutate(
    AmpliconName = str_c(
      "Ots_", tmpname,
      sprintf("%03d", 1:n()), "_", 
      Otsh_v2.0_chromosome, "_", 
      sprintf("%08d",AmpliconStartPos)),
    .before = Otsh_v2.0_chromosome
  ) %>%
  ungroup() %>%
  select(-tmpname)
```

## Deal with the different between ReferenceSequence and TrimmedAmpliconSequence

So, the `RefereneSequence` is sometimes the same as the TrimmedAmpliconSequence, and
sometimes it is not. Mostly it is not the same for the WRAP and LFAR and RoSA, etc.
So, this is not a huge problem.  In cases where things are not the same, the ReferenceSequence
was the sequence I sent to Anthony to design the primers, while the TrimmedAmpliconSequence
is what is actually amplified and what actually gets mapped to the Otsh_V1.0 genome.
So, when we want to figure out the actual position in the genome of particular variants,
we need to know their position in the TrimmedAmpliconSequence, but we actually
only have it in the ReferenceSequence.

So, we have to do some modifications here to make sure we have the position
in the variants in the actualy amplicon sequence.  We can do that pretty
easily.

I am going to rename things a little bit so ReferenceSequence becomes DesignSequence, and
TrimmedAmpliconSequence becomes "AmpliconSequence".

```{r}
# we first need a function to modify the snp locations
# pos is the current list column of positions
# as is the amplicon start position in the reference sequence
new_locations <- function(pos, as, ae) {
  amp_length = ae - as
  pvec <- lapply(str_split(pos, ", *"), as.integer)
  altered_pvec <- map2(.x = pvec, .y = as, .f = function(x, y) x - y + 1)
  thinned_pvec <- map2(.x = altered_pvec, .y = amp_length, .f = function(x, y) x[x > 0 & x <= y])
  ret <- map_chr(.x = thinned_pvec, .f = function(x) paste(x, collapse = ", "))
  unlist(ret)
}

tmp <- final %>%
  mutate(
    ampli_start = str_locate(ReferenceSequence, TrimmedAmpliconSequence)[,1],
    ampli_end = str_locate(ReferenceSequence, TrimmedAmpliconSequence)[,2],
    .before = AmpliconName
  ) %>%
  mutate(
    SNPLocationsInAmpliconSequence = pmap_chr(
      .l = list(pos = SNPLocationsInReferenceSequence, as = ampli_start, ae = ampli_end),
      .f = new_locations
    ),
    .after = SNPLocationsInReferenceSequence
  )

# then, pick out columns in the right order
real_final <- tmp %>% 
  select(
    AmpliconName, 
    Otsh_v2.0_chromosome, 
    AmpliconStartPos, 
    LocusOrigin, 
    OriginalReference, 
    OtherName, 
    ForwardPrimerSequence, 
    ReversePrimerSequence, 
    TrimmedAmpliconSequence,
    SNPLocationsInAmpliconSequence, 
    Otsh_v2.0_coordinates, 
    Alternate_Otsh_v2.0_coordinates, 
    Otsh_v1.0_coordinates, 
    Alternate_Otsh_v1.0_coordinates, 
    MappingNotes,
    ReferenceSequence, 
    SNPLocationsInReferenceSequence, 
  ) %>%
  rename(
    AmpliconSequence = TrimmedAmpliconSequence,
    DesignSequence = ReferenceSequence,
    SNPLocationsInDesignSequence = SNPLocationsInReferenceSequence
  ) %>%
  mutate(
    SNPLocationsInAmpliconSequence = ifelse(SNPLocationsInAmpliconSequence == "NA", NA, SNPLocationsInAmpliconSequence)
  )
```

## Print that dude out

All right!  That pretty much does it.  Let's just print it out.

```{r}
write_csv(real_final, "../../inputs/Calif-Chinook-Amplicon-Panel-Information.csv", na = "")
write_csv(real_final, "../../tex/supp_data/Supp-Data-1-Amplicon-info.csv", na = "")
```
